<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Parallellized Protein Sequence Similarity Calculation based on Sequence Alignment — calcParProtSeqSim • Rcpi</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/spacelab/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">

<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">


<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>

<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Rcpi</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/index.html">Articles</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/road2stat/Rcpi">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

      <div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Parallellized Protein Sequence Similarity Calculation based on Sequence Alignment</h1>
    </div>

    
    <p>Parallellized Protein Sequence Similarity Calculation based on Sequence Alignment</p>
    

    <pre class="usage"><span class='fu'>calcParProtSeqSim</span>(<span class='no'>protlist</span>, <span class='kw'>cores</span> <span class='kw'>=</span> <span class='fl'>2</span>, <span class='kw'>type</span> <span class='kw'>=</span> <span class='st'>"local"</span>,
  <span class='kw'>submat</span> <span class='kw'>=</span> <span class='st'>"BLOSUM62"</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a> Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>protlist</th>
      <td><p>A length <code>n</code> list containing <code>n</code> protein sequences,
each component of the list is a character string,
storing one protein sequence. Unknown sequences should be represented as
<code>''</code>.</p></td>
    </tr>
    <tr>
      <th>cores</th>
      <td><p>Integer. The number of CPU cores to use for parallel execution,
default is <code>2</code>. Users could use the <code>detectCores()</code> function
in the <code>parallel</code> package to see how many cores they could use.</p></td>
    </tr>
    <tr>
      <th>type</th>
      <td><p>Type of alignment, default is <code>'local'</code>,
could be <code>'global'</code> or <code>'local'</code>,
where <code>'global'</code> represents Needleman-Wunsch global alignment;
<code>'local'</code> represents Smith-Waterman local alignment.</p></td>
    </tr>
    <tr>
      <th>submat</th>
      <td><p>Substitution matrix, default is <code>'BLOSUM62'</code>,
could be one of <code>'BLOSUM45'</code>, <code>'BLOSUM50'</code>, <code>'BLOSUM62'</code>,
<code>'BLOSUM80'</code>, <code>'BLOSUM100'</code>, <code>'PAM30'</code>, <code>'PAM40'</code>,
<code>'PAM70'</code>, <code>'PAM120'</code>, <code>'PAM250'</code>.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <code>n</code> x <code>n</code> similarity matrix.</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function implemented the parallellized version for calculating
protein sequence similarity based on sequence alignment.</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <p>See <code>calcTwoProtSeqSim</code> for protein sequence alignment
for two protein sequences. See <code><a href='calcParProtGOSim.html'>calcParProtGOSim</a></code> for
protein similarity calculation based on
Gene Ontology (GO) semantic similarity.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><span class='no'>s1</span> <span class='kw'>=</span> <span class='fu'><a href='readFASTA.html'>readFASTA</a></span>(<span class='fu'>system.file</span>(<span class='st'>'protseq/P00750.fasta'</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>'Rcpi'</span>))<span class='kw'>[[</span><span class='fl'>1</span>]]
<span class='no'>s2</span> <span class='kw'>=</span> <span class='fu'><a href='readFASTA.html'>readFASTA</a></span>(<span class='fu'>system.file</span>(<span class='st'>'protseq/P08218.fasta'</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>'Rcpi'</span>))<span class='kw'>[[</span><span class='fl'>1</span>]]
<span class='no'>s3</span> <span class='kw'>=</span> <span class='fu'><a href='readFASTA.html'>readFASTA</a></span>(<span class='fu'>system.file</span>(<span class='st'>'protseq/P10323.fasta'</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>'Rcpi'</span>))<span class='kw'>[[</span><span class='fl'>1</span>]]
<span class='no'>s4</span> <span class='kw'>=</span> <span class='fu'><a href='readFASTA.html'>readFASTA</a></span>(<span class='fu'>system.file</span>(<span class='st'>'protseq/P20160.fasta'</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>'Rcpi'</span>))<span class='kw'>[[</span><span class='fl'>1</span>]]
<span class='no'>s5</span> <span class='kw'>=</span> <span class='fu'><a href='readFASTA.html'>readFASTA</a></span>(<span class='fu'>system.file</span>(<span class='st'>'protseq/Q9NZP8.fasta'</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>'Rcpi'</span>))<span class='kw'>[[</span><span class='fl'>1</span>]]
<span class='no'>plist</span> <span class='kw'>=</span> <span class='fu'>list</span>(<span class='no'>s1</span>, <span class='no'>s2</span>, <span class='no'>s3</span>, <span class='no'>s4</span>, <span class='no'>s5</span>)

<span class='no'>psimmat</span> <span class='kw'>=</span> <span class='fu'>calcParProtSeqSim</span>(<span class='no'>plist</span>, <span class='kw'>cores</span> <span class='kw'>=</span> <span class='fl'>2</span>, <span class='kw'>type</span> <span class='kw'>=</span> <span class='st'>'local'</span>,
                            <span class='kw'>submat</span> <span class='kw'>=</span> <span class='st'>'BLOSUM62'</span>)
<span class='fu'>print</span>(<span class='no'>psimmat</span>)</pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    
Nan Xiao &lt;<a href = 'https://nanx.me'>https://nanx.me</a>&gt;

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Nan Xiao, Dong-Sheng Cao, Qing-Song Xu.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  </body>
</html>
