<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Compute PSSM (Position-Specific Scoring Matrix) for given protein sequence — extractProtPSSM • Rcpi</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/spacelab/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">

<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">


<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>

<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Rcpi</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/index.html">Articles</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/road2stat/Rcpi">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

      <div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Compute PSSM (Position-Specific Scoring Matrix) for given protein sequence</h1>
    </div>

    
    <p>Compute PSSM (Position-Specific Scoring Matrix) for given protein sequence</p>
    

    <pre class="usage"><span class='fu'>extractProtPSSM</span>(<span class='no'>seq</span>, <span class='kw'>start.pos</span> <span class='kw'>=</span> <span class='fl'>1L</span>, <span class='kw'>end.pos</span> <span class='kw'>=</span> <span class='fu'>nchar</span>(<span class='no'>seq</span>),
  <span class='kw'>psiblast.path</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>makeblastdb.path</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>database.path</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>iter</span> <span class='kw'>=</span> <span class='fl'>5</span>, <span class='kw'>silent</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>evalue</span> <span class='kw'>=</span> <span class='fl'>10L</span>, <span class='kw'>word.size</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>gapopen</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>gapextend</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>matrix</span> <span class='kw'>=</span> <span class='st'>"BLOSUM62"</span>,
  <span class='kw'>threshold</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>seg</span> <span class='kw'>=</span> <span class='st'>"no"</span>, <span class='kw'>soft.masking</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>culling.limit</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>best.hit.overhang</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>best.hit.score.edge</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>xdrop.ungap</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>xdrop.gap</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>xdrop.gap.final</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>window.size</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>gap.trigger</span> <span class='kw'>=</span> <span class='fl'>22L</span>,
  <span class='kw'>num.threads</span> <span class='kw'>=</span> <span class='fl'>1L</span>, <span class='kw'>pseudocount</span> <span class='kw'>=</span> <span class='fl'>0L</span>, <span class='kw'>inclusion.ethresh</span> <span class='kw'>=</span> <span class='fl'>0.002</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a> Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>seq</th>
      <td><p>Character vector, as the input protein sequence.</p></td>
    </tr>
    <tr>
      <th>start.pos</th>
      <td><p>Optional integer denoting the start position of the
fragment window. Default is <code>1</code>,
i.e. the first amino acid of the given sequence.</p></td>
    </tr>
    <tr>
      <th>end.pos</th>
      <td><p>Optional integer denoting the end position of the
fragment window. Default is <code>nchar(seq)</code>,
i.e. the last amino acid of the given sequence.</p></td>
    </tr>
    <tr>
      <th>psiblast.path</th>
      <td><p>Character string indicating the path of the
<code>psiblast</code> program.
If NCBI Blast+ was previously installed in the operation system,
the path will be automatically detected.</p></td>
    </tr>
    <tr>
      <th>makeblastdb.path</th>
      <td><p>Character string indicating the path of the
<code>makeblastdb</code> program.
If NCBI Blast+ was previously installed in the system,
the path will be automatically detected.</p></td>
    </tr>
    <tr>
      <th>database.path</th>
      <td><p>Character string indicating the path of
a reference database (a FASTA file).</p></td>
    </tr>
    <tr>
      <th>iter</th>
      <td><p>Number of iterations to perform for PSI-Blast.</p></td>
    </tr>
    <tr>
      <th>silent</th>
      <td><p>Logical. Whether the PSI-Blast running output
should be shown or not (May not work on some Windows versions and
PSI-Blast versions), default is <code>TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>evalue</th>
      <td><p>Expectation value (E) threshold for saving hits.
Default is <code>10</code>.</p></td>
    </tr>
    <tr>
      <th>word.size</th>
      <td><p>Word size for wordfinder algorithm. An integer &gt;= 2.</p></td>
    </tr>
    <tr>
      <th>gapopen</th>
      <td><p>Integer. Cost to open a gap.</p></td>
    </tr>
    <tr>
      <th>gapextend</th>
      <td><p>Integer. Cost to extend a gap.</p></td>
    </tr>
    <tr>
      <th>matrix</th>
      <td><p>Character string. The scoring matrix name
(default is <code>'BLOSUM62'</code>).</p></td>
    </tr>
    <tr>
      <th>threshold</th>
      <td><p>Minimum word score such that the word is added to
the BLAST lookup table. A real value &gt;= 0.</p></td>
    </tr>
    <tr>
      <th>seg</th>
      <td><p>Character string. Filter query sequence with SEG (<code>'yes'</code>,
<code>'window locut hicut'</code>, or <code>'no'</code> to disable) Default is <code>'no'</code>.</p></td>
    </tr>
    <tr>
      <th>soft.masking</th>
      <td><p>Logical. Apply filtering locations as soft masks?
Default is <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>culling.limit</th>
      <td><p>An integer &gt;= 0. If the query range of a hit is
enveloped by that of at least this many higher-scoring hits,
delete the hit. Incompatible with <code>best.hit.overhang</code> and
<code>best_hit_score_edge</code>.</p></td>
    </tr>
    <tr>
      <th>best.hit.overhang</th>
      <td><p>Best Hit algorithm overhang value
(A real value &gt;= 0 and =&lt; 0.5, recommended value: 0.1).
Incompatible with <code>culling_limit</code>.</p></td>
    </tr>
    <tr>
      <th>best.hit.score.edge</th>
      <td><p>Best Hit algorithm score edge value
(A real value &gt;=0 and =&lt; 0.5, recommended value: 0.1).
Incompatible with <code>culling_limit</code>.</p></td>
    </tr>
    <tr>
      <th>xdrop.ungap</th>
      <td><p>X-dropoff value (in bits) for ungapped extensions.</p></td>
    </tr>
    <tr>
      <th>xdrop.gap</th>
      <td><p>X-dropoff value (in bits) for preliminary gapped extensions.</p></td>
    </tr>
    <tr>
      <th>xdrop.gap.final</th>
      <td><p>X-dropoff value (in bits) for final gapped alignment.</p></td>
    </tr>
    <tr>
      <th>window.size</th>
      <td><p>An integer &gt;= 0. Multiple hits window size,
To specify 1-hit algorithm, use <code>0</code>.</p></td>
    </tr>
    <tr>
      <th>gap.trigger</th>
      <td><p>Number of bits to trigger gapping. Default is <code>22</code>.</p></td>
    </tr>
    <tr>
      <th>num.threads</th>
      <td><p>Integer. Number of threads (CPUs) to use in the
BLAST search. Default is <code>1</code>.</p></td>
    </tr>
    <tr>
      <th>pseudocount</th>
      <td><p>Integer. Pseudo-count value used when constructing PSSM.
Default is <code>0</code>.</p></td>
    </tr>
    <tr>
      <th>inclusion.ethresh</th>
      <td><p>E-value inclusion threshold for pairwise alignments.
Default is <code>0.002</code>.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>The original PSSM, a numeric matrix which has
<code>end.pos - start.pos + 1</code> columns and <code>20</code> named rows.</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function calculates the PSSM (Position-Specific Scoring Matrix) derived
by PSI-Blast for given protein sequence or peptides.
For given protein sequences or peptides, PSSM represents the
log-likelihood of the substitution of the 20 types of amino acids at that
position in the sequence. Note that the output value is not normalized.</p>
    
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>The function requires the <code>makeblastdb</code> and <code>psiblast</code> programs
to be properly installed in the operation system or
their paths provided.</p>
<p>The two command-line programs are included in the NCBI-BLAST+
software package. To install NCBI Blast+, just open the NCBI FTP site
using web browser or FTP software:
<a href = 'ftp://anonymous@ftp.ncbi.nlm.nih.gov:21/blast/executables/blast+/LATEST/'>ftp://anonymous@ftp.ncbi.nlm.nih.gov:21/blast/executables/blast+/LATEST/</a>
then download the executable version of BLAST+ according to your
operation system, and compile or install the downloaded
source code or executable program.</p>
<p>Ubuntu/Debian users can directly use the command
<code>sudo apt-get install ncbi-blast+</code> to install NCBI Blast+.
For OS X users, download <code>ncbi-blast- ... .dmg</code> then install.
For Windows users, download <code>ncbi-blast- ... .exe</code> then install.</p>
    
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Altschul, Stephen F., et al.
"Gapped BLAST and PSI-BLAST: a new generation of protein database search programs."
<em>Nucleic acids research</em> 25.17 (1997): 3389--3402.</p>
<p>Ye, Xugang, Guoli Wang, and Stephen F. Altschul.
"An assessment of substitution scores for protein profile-profile comparison."
<em>Bioinformatics</em> 27.24 (2011): 3356--3363.</p>
<p>Rangwala, Huzefa, and George Karypis.
"Profile-based direct kernels for remote homology detection and fold recognition."
<em>Bioinformatics</em> 21.23 (2005): 4239--4247.</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <p><a href='extractProtPSSMFeature.html'>extractProtPSSMFeature</a> <a href='extractProtPSSMAcc.html'>extractProtPSSMAcc</a></p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><span class='no'>x</span> <span class='kw'>=</span> <span class='fu'><a href='readFASTA.html'>readFASTA</a></span>(<span class='fu'>system.file</span>(<span class='st'>'protseq/P00750.fasta'</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>'Rcpi'</span>))<span class='kw'>[[</span><span class='fl'>1</span>]]

<span class='no'>dbpath</span> <span class='kw'>=</span> <span class='fu'>tempfile</span>(<span class='st'>'tempdb'</span>, <span class='kw'>fileext</span> <span class='kw'>=</span> <span class='st'>'.fasta'</span>)
<span class='fu'>invisible</span>(<span class='fu'>file.copy</span>(<span class='kw'>from</span> <span class='kw'>=</span> <span class='fu'>system.file</span>(<span class='st'>'protseq/Plasminogen.fasta'</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>'Rcpi'</span>), <span class='kw'>to</span> <span class='kw'>=</span> <span class='no'>dbpath</span>))
<span class='no'>pssmmat</span> <span class='kw'>=</span> <span class='fu'>extractProtPSSM</span>(<span class='kw'>seq</span> <span class='kw'>=</span> <span class='no'>x</span>, <span class='kw'>database.path</span> <span class='kw'>=</span> <span class='no'>dbpath</span>)
<span class='fu'>dim</span>(<span class='no'>pssmmat</span>)  <span class='co'># 20 x 562 (P00750: length 562, 20 Amino Acids)</span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>

      <li><a href="#note">Note</a></li>

      <li><a href="#references">References</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    
Nan Xiao &lt;<a href = 'http://nanx.me'>http://nanx.me</a>&gt;

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Nan Xiao, Dongsheng Cao, Qingsong Xu.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  </body>
</html>
