<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Rcpi: R/Bioconductor Package as an Integrated Informatics Platform for Drug Discovery â€¢ Rcpi</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/readable/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Rcpi</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="..//index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Rcpi.html">Get Started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Rcpi-quickref.html">Rcpi Quick Reference Card</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/road2stat/Rcpi">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>Rcpi: R/Bioconductor Package as an Integrated Informatics Platform for Drug Discovery</h1>
                        <h4 class="author">Nan Xiao &lt;<a href="https://nanx.me" class="uri">https://nanx.me</a>&gt;</h4>
            
            <h4 class="date">2017-09-24</h4>
          </div>

    
    
<div class="contents">
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>The Rcpi package <span class="citation">Cao <em>et al.</em> (2015)</span> offers an R/Bioconductor package emphasizing the comprehensive integration of bioinformatics and chemoinformatics into a molecular informatics platform for drug discovery.</p>
<p>Rcpi implemented and integrated the state-of-the-art protein sequence descriptors and molecular descriptors/fingerprints with R. For protein sequences, the Rcpi package can</p>
<ul>
<li>Calculate six protein descriptor groups composed of fourteen types of commonly used structural and physicochemical descriptors that include 9,920 descriptors.</li>
<li>Calculate profile-based protein representation derived by PSSM (Position-Specific Scoring Matrix).</li>
<li>Calculate six types of generalized scales-based descriptors derived by various dimensionality reduction methods for proteochemometric (PCM) modeling.</li>
<li>Parallellized pairwise similarity computation derived by protein sequence alignment and Gene Ontology (GO) semantic similarity measures within a list of proteins.</li>
</ul>
<p>For small molecules, the Rcpi package can</p>
<ul>
<li>Calculate 307 molecular descriptors (2D/3D), including constitutional, topological, geometrical, and electronic descriptors, etc.</li>
<li>Calculate more than ten types of molecular fingerprints, including FP4 keys, E-state fingerprints, MACCS keys, etc., and parallelized chemical similarity search.</li>
<li>Parallelized pairwise similarity computation derived by fingerprints and maximum common substructure search within a list of small molecules.</li>
</ul>
<p>By combining various types of descriptors for drugs and proteins in different methods, interaction descriptors representing protein-protein or compound-protein interactions can be conveniently generated with Rcpi, including:</p>
<ul>
<li>Two types of compound-protein interaction (CPI) descriptors</li>
<li>Three types of protein-protein interaction (PPI) descriptors</li>
</ul>
<p>Several useful auxiliary utilities are also shipped with Rcpi:</p>
<ul>
<li>Parallelized molecule and protein sequence retrieval from several online databases, like PubChem, ChEMBL, KEGG, DrugBank, UniProt, RCSB PDB, etc.</li>
<li>Loading molecules stored in SMILES/SDF files and loading protein sequences from FASTA/PDB files</li>
<li>Molecular file format conversion</li>
</ul>
<p>The computed protein sequence descriptors, molecular descriptors/fingerprints, interaction descriptors and pairwise similarities are widely used in various research fields relevant to drug disvery, such as, bioinformatics, chemoinformatics, proteochemometrics and chemogenomics.</p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To install the Rcpi package, use:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(<span class="st">"https://bioconductor.org/biocLite.R"</span>)
<span class="kw">biocLite</span>(<span class="st">"Rcpi"</span>)</code></pre></div>
<p>To make the Rcpi package fully functional (especially the Open Babel related functionalities), we recommend the users also install the <em>Enhances</em> packages by:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(<span class="st">"https://bioconductor.org/biocLite.R"</span>)
<span class="kw">biocLite</span>(<span class="st">"Rcpi"</span>, <span class="dt">dependencies =</span> <span class="kw">c</span>(<span class="st">"Imports"</span>, <span class="st">"Enhances"</span>))</code></pre></div>
<p>Several dependencies of the Rcpi package may require some system-level libraries, check the corresponding manuals of these packages for detailed installation guides.</p>
</div>
<div id="how-to-cite-rcpi" class="section level2">
<h2 class="hasAnchor">
<a href="#how-to-cite-rcpi" class="anchor"></a>How to Cite Rcpi</h2>
<p>If you feel Rcpi is useful in your research, please cite our paper:</p>
<blockquote>
Dong-Sheng Cao, Nan Xiao, Qing-Song Xu, and Alex F. Chen. (2015). Rcpi: R/Bioconductor package to generate various descriptors of proteins, compounds and their interactions. <em>Bioinformatics</em> 31 (2), 279-281.
</blockquote>
<p>BibTeX entry:</p>
<div class="sourceCode"><pre class="sourceCode bibtex"><code class="sourceCode bibtex"><span class="va">@article</span>{<span class="ot">Rcpi2015</span>,
<span class="dt">author</span> = {Cao, Dong-Sheng and Xiao, Nan and Xu, Qing-Song and Alex F. Chen.},
<span class="dt">title</span> = {{Rcpi: R/Bioconductor package to generate various descriptors
  of proteins, compounds and their interactions}},
<span class="dt">journal</span> = {Bioinformatics},
<span class="dt">year</span> = {2015},
<span class="dt">volume</span> = {31},
<span class="dt">number</span> = {2},
<span class="dt">pages</span> = {279--281},
<span class="dt">doi</span> = {10.1093/bioinformatics/btu624},
<span class="dt">issn</span> = {1367-4803},
<span class="dt">url</span> = {http://bioinformatics.oxfordjournals.org/content/31/2/279}
}</code></pre></div>
</div>
</div>
<div id="applications-in-bioinformatics" class="section level1">
<h1 class="hasAnchor">
<a href="#applications-in-bioinformatics" class="anchor"></a>Applications in Bioinformatics</h1>
<p>For bioinformatics research, Rcpi calculates commonly used descriptors and proteochemometric (PCM) modeling descriptors for protein sequences. Rcpi also computes pairwise similarities derived by GO semantic similarity and sequence alignment.</p>
<div id="predicting-protein-subcellular-localization" class="section level2">
<h2 class="hasAnchor">
<a href="#predicting-protein-subcellular-localization" class="anchor"></a>Predicting Protein Subcellular Localization</h2>
<p>Protein subcellular localization prediction involves the computational prediction of where a protein resides in a cell. It is an important component of bioinformatics-based prediction of protein function and genome annotation, and can also aid us to identify novel drug targets.</p>
<p>Here we use the subcellular localization dataset of human proteins presented in the study of <span class="citation">Chou and Shen (2008)</span> for a demonstration. The complete dataset includes 3,134 protein sequences (2,750 different proteins), classified into 14 human subcellular locations. We selected two classes of proteins as our benchmark dataset. Class 1 contains 325 <em>extracell</em> proteins, and class 2 includes 307 <em>mitochondrion</em> proteins.</p>
<p>First, we load the Rcpi package, then read the protein sequences stored in two separated FASTA files with <code><a href="../reference/readFASTA.html">readFASTA()</a></code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"Rcpi"</span>)

<span class="co"># load FASTA files</span>
extracell =<span class="st"> </span><span class="kw"><a href="../reference/readFASTA.html">readFASTA</a></span>(<span class="kw">system.file</span>(
  <span class="st">"vignettedata/extracell.fasta"</span>, <span class="dt">package =</span> <span class="st">"Rcpi"</span>))
mitonchon =<span class="st"> </span><span class="kw"><a href="../reference/readFASTA.html">readFASTA</a></span>(<span class="kw">system.file</span>(
  <span class="st">"vignettedata/mitochondrion.fasta"</span>, <span class="dt">package =</span> <span class="st">"Rcpi"</span>))</code></pre></div>
<p>The loaded sequences are stored as two lists in R, and each component in the list is a character string representing one protein sequence. In this case, there are 325 <em>extracell</em> protein sequences and 306 <em>mitonchon</em> protein sequences:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">length</span>(extracell)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## [1] 325</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">length</span>(mitonchon)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## [1] 306</code></pre></div>
<p>To assure that the protein sequences only have the twenty standard amino acid types which is required for the descriptor computation, we use the <code><a href="../reference/checkProt.html">checkProt()</a></code> function to do the amino acid type sanity checking and remove the <em>non-standard</em> protein sequences:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">extracell =<span class="st"> </span>extracell[(<span class="kw">sapply</span>(extracell, checkProt))]
mitonchon =<span class="st"> </span>mitonchon[(<span class="kw">sapply</span>(mitonchon, checkProt))]</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">length</span>(extracell)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## [1] 323</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">length</span>(mitonchon)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## [1] 304</code></pre></div>
<p>Two protein sequences were removed from each class. For the remaining sequences, we calculate the amphiphilic pseudo amino acid composition (APAAC) descriptor <span class="citation">Chou (2005)</span> and create class labels for the random forest classification modeling.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># calculate APAAC descriptors</span>
x1 =<span class="st"> </span><span class="kw">t</span>(<span class="kw">sapply</span>(extracell, extractProtAPAAC))
x2 =<span class="st"> </span><span class="kw">t</span>(<span class="kw">sapply</span>(mitonchon, extractProtAPAAC))
x  =<span class="st"> </span><span class="kw">rbind</span>(x1, x2)

<span class="co"># make class labels</span>
labels =<span class="st"> </span><span class="kw">as.factor</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>, <span class="kw">length</span>(extracell)), <span class="kw">rep</span>(<span class="dv">1</span>, <span class="kw">length</span>(mitonchon))))</code></pre></div>
<p>In Rcpi, the functions of commonly used descriptors for protein sequences and proteochemometric (PCM) modeling descriptors are named after <code>extractProt...()</code> and <code>extractPCM...()</code>.</p>
<p>Next, we will split the data into a 75% training set and a 25% test set.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">set.seed</span>(<span class="dv">1001</span>)

<span class="co"># split training and test set</span>
tr.idx =<span class="st"> </span><span class="kw">c</span>(
  <span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(x1), <span class="kw">round</span>(<span class="kw">nrow</span>(x1) <span class="op">*</span><span class="st"> </span><span class="fl">0.75</span>)),
  <span class="kw">sample</span>(<span class="kw">nrow</span>(x1) <span class="op">+</span><span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(x2), <span class="kw">round</span>(<span class="kw">nrow</span>(x2) <span class="op">*</span><span class="st"> </span><span class="fl">0.75</span>)))
te.idx =<span class="st"> </span><span class="kw">setdiff</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(x), tr.idx)

x.tr   =<span class="st"> </span>x[tr.idx, ]
x.te   =<span class="st"> </span>x[te.idx, ]
y.tr   =<span class="st"> </span>labels[tr.idx]
y.te   =<span class="st"> </span>labels[te.idx]</code></pre></div>
<p>We will train a random forest classification model on the training set with 5-fold cross-validation, using the  package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"randomForest"</span>)
rf.fit =<span class="st"> </span><span class="kw">randomForest</span>(x.tr, y.tr, <span class="dt">cv.fold =</span> <span class="dv">5</span>)
<span class="kw">print</span>(rf.fit)</code></pre></div>
<p>The training result is:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Call:
##  randomForest(x = x.tr, y = y.tr, cv.fold = 5)
##                Type of random forest: classification
##                      Number of trees: 500
## No. of variables tried at each split: 8
##
##         OOB estimate of  error rate: 25.11%
## Confusion matrix:
##     0   1 class.error
## 0 196  46   0.1900826
## 1  72 156   0.3157895</code></pre></div>
<p>With the model trained on the training set, we predict on the test set and plot the ROC curve with the  package, as is shown in Figure 1.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># predict on test set</span>
rf.pred =<span class="st"> </span><span class="kw">predict</span>(rf.fit, <span class="dt">newdata =</span> x.te, <span class="dt">type =</span> <span class="st">"prob"</span>)[, <span class="dv">1</span>]

<span class="co"># plot ROC curve</span>
<span class="kw">library</span>(<span class="st">"pROC"</span>)
<span class="kw"><a href="http://www.rdocumentation.org/packages/pROC/topics/plot.roc">plot.roc</a></span>(y.te, rf.pred, <span class="dt">grid =</span> <span class="ot">TRUE</span>, <span class="dt">print.auc =</span> <span class="ot">TRUE</span>)</code></pre></div>
<img src="figures/ex1-1.png" width="65%" class="img-responsive" align="center" alt="ROC curve for the test set of protein subcellular localization data"><p align="center">
Figure 1: ROC curve for the test set of protein subcellular localization data
</p>
<p>The area under the ROC curve (AUC) is:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Call:
## plot.roc.default(x = y.te, predictor = rf.pred, col = "#0080ff",
##                  grid = TRUE, print.auc = TRUE)
##
## Data: rf.pred in 81 controls (y.te 0) &gt; 76 cases (y.te 1).
## Area under the curve: 0.8697</code></pre></div>
</div>
</div>
<div id="applications-in-chemoinformatics" class="section level1">
<h1 class="hasAnchor">
<a href="#applications-in-chemoinformatics" class="anchor"></a>Applications in Chemoinformatics</h1>
<p>For chemoinformatics research, Rcpi calculates various types of molecular descriptors/fingerprints, and computes pairwise similarities derived by fingerprints and maximum common substructure search. Rcpi also provides the <code><a href="../reference/searchDrug.html">searchDrug()</a></code> function for parallelized molecular similarity search based on these similarity types.</p>
<div id="regression-modeling-in-qsrr-study-of-retention-indices" class="section level2">
<h2 class="hasAnchor">
<a href="#regression-modeling-in-qsrr-study-of-retention-indices" class="anchor"></a>Regression Modeling in QSRR Study of Retention Indices</h2>
<p>In <span class="citation">Yan <em>et al.</em> (2012)</span>, a quantitative structure-retention relationship study was performed for 656 flavor compounds on four stationary phases of different polarities, using constitutional, topological and geometrical molecular descriptors. The gas chromatographic retention indices (RIs) of these compounds were accurately predicted using linear models. Here we choose the molecules and their RIs of one stationary phase (OV101) as our benchmark dataset.</p>
<p>Since it would be rather tedious to implement the complete cross-validation procedures, the R package <code>caret</code> is used here. To run the R code below, users need to install the  package and the required predictive modeling packages first. The  package provides a unified interface for the modeling tuning task across different statistical machine learning packages. It is particularly helpful in QSAR modeling, for it contains tools for data splitting, pre-processing, feature selection, model tuning and other functionalities.</p>
<p>Just like the last section, we load the Rcpi package, and read the molecules stored in a SMILES file:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"Rcpi"</span>)

RI.smi =<span class="st"> </span><span class="kw">system.file</span>(
  <span class="st">"vignettedata/FDAMDD.smi"</span>, <span class="dt">package =</span> <span class="st">"Rcpi"</span>)
RI.csv =<span class="st"> </span><span class="kw">system.file</span>(
  <span class="st">"vignettedata/RI.csv"</span>, <span class="dt">package =</span> <span class="st">"Rcpi"</span>)

x.mol =<span class="st"> </span><span class="kw"><a href="../reference/readMolFromSmi.html">readMolFromSmi</a></span>(RI.smi, <span class="dt">type =</span> <span class="st">"mol"</span>)
x.tab =<span class="st"> </span><span class="kw">read.table</span>(RI.csv, <span class="dt">sep =</span> <span class="st">"</span><span class="ch">\t</span><span class="st">"</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)
y =<span class="st"> </span>x.tab<span class="op">$</span>RI</code></pre></div>
<p>The <code><a href="../reference/readMolFromSmi.html">readMolFromSmi()</a></code> function is used for reading molecules from SMILES files, for molecules stored in SDF files, use <code><a href="../reference/readMolFromSDF.html">readMolFromSDF()</a></code> instead.</p>
<p>The CSV file <code>RI.csv</code> contains tabular data for the retention indices, compound name, and odor information of the compounds. Here we only extracted the RI values by calling <code>x.tab$RI</code>.</p>
<p>After the molecules were properly loaded, we calculate several selected molecular descriptors. The corresponding functions for molecular descriptor calculation are all named after <code>extractDrug...()</code> in <code>Rcpi</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># calculate selected molecular descriptors</span>
x =<span class="st"> </span><span class="kw">suppressWarnings</span>(<span class="kw">cbind</span>(
    <span class="kw"><a href="../reference/extractDrugALOGP.html">extractDrugALOGP</a></span>(x.mol),
    <span class="kw"><a href="../reference/extractDrugApol.html">extractDrugApol</a></span>(x.mol),
    <span class="kw"><a href="../reference/extractDrugECI.html">extractDrugECI</a></span>(x.mol),
    <span class="kw"><a href="../reference/extractDrugTPSA.html">extractDrugTPSA</a></span>(x.mol),
    <span class="kw"><a href="../reference/extractDrugWeight.html">extractDrugWeight</a></span>(x.mol),
    <span class="kw"><a href="../reference/extractDrugWienerNumbers.html">extractDrugWienerNumbers</a></span>(x.mol),
    <span class="kw"><a href="../reference/extractDrugZagrebIndex.html">extractDrugZagrebIndex</a></span>(x.mol)))</code></pre></div>
<p>After the descriptors were calculated, the result <code>x</code> will be a data frame, each row represents one molecule, and each column is one descriptor (predictor). The Rcpi package integrated the molecular descriptors and chemical fingerprints calculated by the <code>rcdk</code> package <span class="citation">Steinbeck <em>et al.</em> (2003)</span> and the ChemmineOB package <span class="citation">Horan and Girke (2013)</span>.</p>
<p>Next, a partial least squares model will be fitted with the <code>pls</code> and the <code>caret</code> package. The cross-validation setting is 5-fold repeated cross-validation (repeat for 10 times).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># regression on training set</span>
<span class="kw">library</span>(<span class="st">"caret"</span>)
<span class="kw">library</span>(<span class="st">"pls"</span>)

<span class="co"># cross-validation settings</span>
ctrl =<span class="st"> </span><span class="kw">trainControl</span>(
  <span class="dt">method =</span> <span class="st">"repeatedcv"</span>, <span class="dt">number =</span> <span class="dv">5</span>, <span class="dt">repeats =</span> <span class="dv">10</span>,
  <span class="dt">summaryFunction =</span> defaultSummary)

<span class="co"># train a pls model</span>
<span class="kw">set.seed</span>(<span class="dv">1002</span>)
pls.fit =<span class="st"> </span><span class="kw">train</span>(
  x, y, <span class="dt">method =</span> <span class="st">"pls"</span>, <span class="dt">tuneLength =</span> <span class="dv">10</span>, <span class="dt">trControl =</span> ctrl,
  <span class="dt">metric =</span> <span class="st">"RMSE"</span>, <span class="dt">preProc =</span> <span class="kw">c</span>(<span class="st">"center"</span>, <span class="st">"scale"</span>))

<span class="co"># print cross-validation result</span>
<span class="kw">print</span>(pls.fit)</code></pre></div>
<p>The cross-validation result is:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Partial Least Squares
##
## 297 samples
##  10 predictors
##
## Pre-processing: centered, scaled
## Resampling: Cross-Validated (5 fold, repeated 10 times)
##
## Summary of sample sizes: 237, 237, 237, 238, 239, 238, ...
##
## Resampling results across tuning parameters:
##
##   ncomp  RMSE  Rsquared  RMSE SD  Rsquared SD
##   1      104   0.884     9.44     0.0285
##   2      86.4  0.92      6.99     0.0194
##   3      83.8  0.924     6.56     0.0185
##   4      79.6  0.931     6.98     0.0194
##   5      76.3  0.937     7.45     0.0187
##   6      74.7  0.94      6.85     0.0162
##   7      73.7  0.941     6.75     0.0159
##   8      73.5  0.942     6.5      0.0142
##   9      72.5  0.944     6.18     0.0137
##
## RMSE was used to select the optimal model using  the smallest value.
## The final value used for the model was ncomp = 9.</code></pre></div>
<p>We see that the RMSE of the PLS regression model was decreasing when the number of principal components (<code>ncomp</code>) was increasing. We can plot the components and RMSE to helps us select the desired number of principal components used in the model.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># # of components vs RMSE</span>
<span class="kw">print</span>(<span class="kw">plot</span>(pls.fit, <span class="dt">asp =</span> <span class="fl">0.5</span>))</code></pre></div>
<img src="figures/ex2-1.png" width="85%" class="img-responsive" align="center" alt="Number of principal components vs. RMSE for the PLS regression model"><p align="center">
Figure 2: Number of principal components vs.Â RMSE for the PLS regression model
</p>
<p>From Figure 2, we consider that selecting six or seven components is acceptable. At last, we plot the experimental RIs and the predicted RIs to see if the model fits well on the training set (Figure 3):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># plot experimental RIs vs predicted RIs</span>
<span class="kw">plot</span>(y, <span class="kw">predict</span>(pls.fit, x), <span class="dt">xlim =</span> <span class="kw">range</span>(y), <span class="dt">ylim =</span> <span class="kw">range</span>(y),
     <span class="dt">xlab =</span> <span class="st">"Experimental RIs"</span>, <span class="dt">ylab =</span> <span class="st">"Predicted RIs"</span>)
<span class="kw">abline</span>(<span class="dt">a =</span> <span class="dv">0</span>, <span class="dt">b =</span> <span class="dv">1</span>)</code></pre></div>
<img src="figures/ex2-2.png" width="65%" class="img-responsive" align="center" alt="Experimental RIs vs. Predicted RIs"><p align="center">
Figure 3: Experimental RIs vs.Â Predicted RIs
</p>
</div>
<div id="in-silico-toxicity-classification-for-drug-discovery" class="section level2">
<h2 class="hasAnchor">
<a href="#in-silico-toxicity-classification-for-drug-discovery" class="anchor"></a><em>In Silico</em> Toxicity Classification for Drug Discovery</h2>
<p>In the perspective of quantitative pharmacology, the successful discovery of novel drugs depends on the pharmacokinetics properties, like <strong>a</strong>bsorption, <strong>d</strong>istribution, <strong>m</strong>etabolism, and <strong>e</strong>xcretion. In addition, the potential <strong>t</strong>oxicity of chemical compounds is taken into account. QSAR or QSPR methods are usually employed to predict the ADME/T qualities of potential drug candidates.</p>
<p>In <span class="citation">D-S Cao <em>et al.</em> (2012)</span>, quantitative structure-toxicity relationship (QSTR) models were established for classifying five toxicity datasets. Here we use the maximum recommended daily dose dataset (FDAMDD) from FDA Center for Drug Evaluation and Research as the benchmark dataset.</p>
<p>First, load the drug molecules stored in a SMILES file into R:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"Rcpi"</span>)

fdamdd.smi =<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"vignettedata/FDAMDD.smi"</span>, <span class="dt">package =</span> <span class="st">"Rcpi"</span>)
fdamdd.csv =<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"vignettedata/FDAMDD.csv"</span>, <span class="dt">package =</span> <span class="st">"Rcpi"</span>)

x.mol =<span class="st"> </span><span class="kw"><a href="../reference/readMolFromSmi.html">readMolFromSmi</a></span>(fdamdd.smi, <span class="dt">type =</span> <span class="st">"mol"</span>)
x.smi =<span class="st"> </span><span class="kw"><a href="../reference/readMolFromSmi.html">readMolFromSmi</a></span>(fdamdd.smi, <span class="dt">type =</span> <span class="st">"text"</span>)
y =<span class="st"> </span><span class="kw">as.factor</span>(<span class="kw">paste0</span>(<span class="st">"class"</span>, <span class="kw">scan</span>(fdamdd.csv)))</code></pre></div>
<p>The object <code>x.mol</code> is used for computing the MACCS and E-state fingerprints; the object <code>x.smi</code> is used for computing the FP4 fingerprints. The 0-1 class labels stored in file <code>FDAMDD.csv</code> indicates whether the drug molecule has high toxicity or not.</p>
<p>Then we calculate three different types of molecular fingerprints (E-state, MACCS, and FP4) for the drug molecules:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># calculate molecular fingerprints</span>
x1 =<span class="st"> </span><span class="kw"><a href="../reference/extractDrugEstateComplete.html">extractDrugEstateComplete</a></span>(x.mol)
x2 =<span class="st"> </span><span class="kw"><a href="../reference/extractDrugMACCSComplete.html">extractDrugMACCSComplete</a></span>(x.mol)
x3 =<span class="st"> </span><span class="kw"><a href="../reference/extractDrugOBFP4.html">extractDrugOBFP4</a></span>(x.smi, <span class="dt">type =</span> <span class="st">"smile"</span>)</code></pre></div>
<p>As the nature of fingerprint-based structure representation, the calculated 0-1 matrix <code>x1</code>, <code>x2</code>, and <code>x3</code> will be very sparse. Since there are several columns have nearly exactly the same value for all the molecules, we should remove them with <code>nearZeroVar()</code> in <code>caret</code> before modeling, and split our training set and test set:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"caret"</span>)

<span class="co"># remove near-zero variance variables</span>
x1 =<span class="st"> </span>x1[, <span class="op">-</span><span class="kw">nearZeroVar</span>(x1)]
x2 =<span class="st"> </span>x2[, <span class="op">-</span><span class="kw">nearZeroVar</span>(x2)]
x3 =<span class="st"> </span>x3[, <span class="op">-</span><span class="kw">nearZeroVar</span>(x3)]

<span class="co"># split training and test set</span>
<span class="kw">set.seed</span>(<span class="dv">1003</span>)
tr.idx =<span class="st"> </span><span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(x1), <span class="kw">round</span>(<span class="kw">nrow</span>(x1) <span class="op">*</span><span class="st"> </span><span class="fl">0.75</span>))
te.idx =<span class="st"> </span><span class="kw">setdiff</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(x1), tr.idx)
x1.tr  =<span class="st"> </span>x1[tr.idx, ]
x1.te  =<span class="st"> </span>x1[te.idx, ]
x2.tr  =<span class="st"> </span>x2[tr.idx, ]
x2.te  =<span class="st"> </span>x2[te.idx, ]
x3.tr  =<span class="st"> </span>x3[tr.idx, ]
x3.te  =<span class="st"> </span>x3[te.idx, ]
y.tr   =<span class="st"> </span>y[tr.idx]
y.te   =<span class="st"> </span>y[te.idx]</code></pre></div>
<p>On the training sets, we will train three classification models separately using SVM (RBF kernel), using the <code>kernlab</code> package and the <code>caret</code> package. The cross-validation setting is 5-fold repeated CV (repeat for 10 times).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># svm classification on training sets</span>
<span class="kw">library</span>(<span class="st">"kernlab"</span>)

<span class="co"># cross-validation settings</span>
ctrl =<span class="st"> </span><span class="kw">trainControl</span>(<span class="dt">method =</span> <span class="st">"cv"</span>, <span class="dt">number =</span> <span class="dv">5</span>, <span class="dt">repeats =</span> <span class="dv">10</span>,
                    <span class="dt">classProbs =</span> <span class="ot">TRUE</span>,
                    <span class="dt">summaryFunction =</span> twoClassSummary)

<span class="co"># SVM with RBF kernel</span>
svm.fit1 =<span class="st"> </span><span class="kw">train</span>(
  x1.tr, y.tr, <span class="dt">method =</span> <span class="st">"svmRadial"</span>, <span class="dt">trControl =</span> ctrl,
  <span class="dt">metric =</span> <span class="st">"ROC"</span>, <span class="dt">preProc =</span> <span class="kw">c</span>(<span class="st">"center"</span>, <span class="st">"scale"</span>))
svm.fit2 =<span class="st"> </span><span class="kw">train</span>(
  x2.tr, y.tr, <span class="dt">method =</span> <span class="st">"svmRadial"</span>, <span class="dt">trControl =</span> ctrl,
  <span class="dt">metric =</span> <span class="st">"ROC"</span>, <span class="dt">preProc =</span> <span class="kw">c</span>(<span class="st">"center"</span>, <span class="st">"scale"</span>))
svm.fit3 =<span class="st"> </span><span class="kw">train</span>(
  x3.tr, y.tr, <span class="dt">method =</span> <span class="st">"svmRadial"</span>, <span class="dt">trControl =</span> ctrl,
  <span class="dt">metric =</span> <span class="st">"ROC"</span>, <span class="dt">preProc =</span> <span class="kw">c</span>(<span class="st">"center"</span>, <span class="st">"scale"</span>))

<span class="co"># print cross-validation result</span>
<span class="kw">print</span>(svm.fit1)
<span class="kw">print</span>(svm.fit2)
<span class="kw">print</span>(svm.fit3)</code></pre></div>
<p>The training result when using E-state fingerprints:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Support Vector Machines with Radial Basis Function Kernel
##
## 597 samples
##  23 predictors
##   2 classes: "class0", "class1"
##
## Pre-processing: centered, scaled
## Resampling: Cross-Validated (5 fold)
##
## Summary of sample sizes: 478, 479, 477, 477, 477
##
## Resampling results across tuning parameters:
##
##   C     ROC    Sens   Spec   ROC SD  Sens SD  Spec SD
##   0.25  0.797  0.7    0.765  0.0211  0.0442   0.00666
##   0.5   0.808  0.696  0.79   0.0173  0.059    0.0236
##   1     0.812  0.703  0.781  0.0191  0.0664   0.0228
##
## Tuning parameter "sigma" was held constant at a value of 0.02921559
## ROC was used to select the optimal model using  the largest value.
## The final values used for the model were sigma = 0.0292 and C = 1.</code></pre></div>
<p>We can see that, after removing the near zero variance predictors, there were only 23 predictors left for the original 79 E-state fingerprints.</p>
<p>The training result when using MACCS keys:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Support Vector Machines with Radial Basis Function Kernel
##
## 597 samples
## 126 predictors
##   2 classes: "class0", "class1"
##
## Pre-processing: centered, scaled
## Resampling: Cross-Validated (5 fold)
##
## Summary of sample sizes: 477, 477, 477, 478, 479
##
## Resampling results across tuning parameters:
##
##   C     ROC    Sens   Spec   ROC SD  Sens SD  Spec SD
##   0.25  0.834  0.715  0.775  0.0284  0.0994   0.0589
##   0.5   0.848  0.726  0.79   0.0299  0.065    0.0493
##   1     0.863  0.769  0.793  0.0307  0.0229   0.0561
##
## Tuning parameter "sigma" was held constant at a value of 0.004404305
## ROC was used to select the optimal model using  the largest value.
## The final values used for the model were sigma = 0.0044 and C = 1.</code></pre></div>
<p>There are 126 predictors left for the original 166 MACCS keys after removing the near-zero variance predictors. The model performance by AUC values is slightly better than using the E-state fingerprints.</p>
<p>The training result when using FP4 fingerprints:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Support Vector Machines with Radial Basis Function Kernel
##
## 597 samples
##  58 predictors
##   2 classes: "class0", "class1"
##
## Pre-processing: centered, scaled
## Resampling: Cross-Validated (5 fold)
##
## Summary of sample sizes: 478, 478, 477, 478, 477
##
## Resampling results across tuning parameters:
##
##   C     ROC    Sens   Spec   ROC SD  Sens SD  Spec SD
##   0.25  0.845  0.769  0.746  0.0498  0.0458   0.0877
##   0.5   0.856  0.744  0.777  0.0449  0.0148   0.0728
##   1     0.863  0.751  0.777  0.0428  0.036    0.0651
##
## Tuning parameter "sigma" was held constant at a value of 0.01077024
## ROC was used to select the optimal model using  the largest value.
## The final values used for the model were sigma = 0.0108 and C = 1.</code></pre></div>
<p>There are 58 predictors left for the original 512 FP4 fingerprints after the screening. The model performance by AUC values is almost the same comparing to using MACCS keys, and better than using E-state fingerprints.</p>
<p>We predict on the test sets with the established models, and plot the ROC curves in one figure, as is shown in Figure 4.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># predict on test set</span>
svm.pred1 =<span class="st"> </span><span class="kw">predict</span>(svm.fit1, <span class="dt">newdata =</span> x1.te, <span class="dt">type =</span> <span class="st">"prob"</span>)[, <span class="dv">1</span>]
svm.pred2 =<span class="st"> </span><span class="kw">predict</span>(svm.fit2, <span class="dt">newdata =</span> x2.te, <span class="dt">type =</span> <span class="st">"prob"</span>)[, <span class="dv">1</span>]
svm.pred3 =<span class="st"> </span><span class="kw">predict</span>(svm.fit3, <span class="dt">newdata =</span> x3.te, <span class="dt">type =</span> <span class="st">"prob"</span>)[, <span class="dv">1</span>]

<span class="co"># generate colors</span>
<span class="kw">library</span>(<span class="st">"RColorBrewer"</span>)
pal =<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/RColorBrewer/topics/ColorBrewer">brewer.pal</a></span>(<span class="dv">3</span>, <span class="st">"Set1"</span>)

<span class="co"># ROC curves of different fingerprints</span>
<span class="kw">library</span>(<span class="st">"pROC"</span>)
<span class="kw">plot</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/pROC/topics/smooth">smooth</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/pROC/topics/roc">roc</a></span>(y.te, svm.pred1)), <span class="dt">col =</span> pal[<span class="dv">1</span>], <span class="dt">grid =</span> <span class="ot">TRUE</span>)
<span class="kw">plot</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/pROC/topics/smooth">smooth</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/pROC/topics/roc">roc</a></span>(y.te, svm.pred2)), <span class="dt">col =</span> pal[<span class="dv">2</span>], <span class="dt">grid =</span> <span class="ot">TRUE</span>, <span class="dt">add =</span> <span class="ot">TRUE</span>)
<span class="kw">plot</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/pROC/topics/smooth">smooth</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/pROC/topics/roc">roc</a></span>(y.te, svm.pred3)), <span class="dt">col =</span> pal[<span class="dv">3</span>], <span class="dt">grid =</span> <span class="ot">TRUE</span>, <span class="dt">add =</span> <span class="ot">TRUE</span>)</code></pre></div>
<img src="figures/ex3-1.png" width="65%" class="img-responsive" align="center" alt="Smoothed ROC curves for different fingerprint types"><p align="center">
Figure 4: Smoothed ROC curves for different fingerprint types
</p>
</div>
<div id="clustering-of-molecules-based-on-structural-similarities" class="section level2">
<h2 class="hasAnchor">
<a href="#clustering-of-molecules-based-on-structural-similarities" class="anchor"></a>Clustering of Molecules Based on Structural Similarities</h2>
<p>Apart from supervised methods (classification and regression), unsupervised approaches, like clustering, is also widely applied in the quantitative research of drugs.</p>
<p>In reality, there are usually too many chemical compounds available for identifying drug-like molecules. Thus it would be attractive using clustering methods to aid the selection of a representative subset of all available compounds. For a clustering approach that groups compounds together by their structural similarity, applying the principle <em>similar compounds have similar properties</em> means that we only need to test the representative compounds from each individual cluster, rather than do the time-consuming complete set of experiments, and this should be sufficient to understand the structure-activity relationships of the whole compound set.</p>
<p>The Rcpi package provides easy-to-use functions for computing the similarity between small molecules derived by molecular fingerprints and maximum common substructure search.</p>
<p>As a example, the SDF file <code>tyrphostin.sdf</code> below is a database composed by searching <em>tyrphostin</em> in PubChem and filtered by _Lipinskiâ€œs rule of five_. We load this SDF file using <code><a href="../reference/readMolFromSDF.html">readMolFromSDF()</a></code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"Rcpi"</span>)
mols =<span class="st"> </span><span class="kw"><a href="../reference/readMolFromSDF.html">readMolFromSDF</a></span>(<span class="kw">system.file</span>(
    <span class="st">"compseq/tyrphostin.sdf"</span>, <span class="dt">package =</span> <span class="st">"Rcpi"</span>))</code></pre></div>
<p>Then compute the E-state fingerprints for all the molecules using <code><a href="../reference/extractDrugEstate.html">extractDrugEstate()</a></code>, and calculate their pairwise similarity matrix with <code><a href="../reference/calcDrugFPSim.html">calcDrugFPSim()</a></code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">simmat =<span class="st"> </span><span class="kw">diag</span>(<span class="kw">length</span>(mols))

<span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(mols)) {
 <span class="cf">for</span> (j <span class="cf">in</span> i<span class="op">:</span><span class="kw">length</span>(mols)) {
   fp1 =<span class="st"> </span><span class="kw"><a href="../reference/extractDrugEstate.html">extractDrugEstate</a></span>(mols[[i]])
   fp2 =<span class="st"> </span><span class="kw"><a href="../reference/extractDrugEstate.html">extractDrugEstate</a></span>(mols[[j]])
   tmp =<span class="st"> </span><span class="kw"><a href="../reference/calcDrugFPSim.html">calcDrugFPSim</a></span>(fp1, fp2, <span class="dt">fptype =</span> <span class="st">"compact"</span>, <span class="dt">metric =</span> <span class="st">"tanimoto"</span>)
   simmat[i, j] =<span class="st"> </span>tmp
   simmat[j, i] =<span class="st"> </span>tmp
 }
}</code></pre></div>
<p>For the computed similarity matrix <code>simmat</code>, we will try to do hierarchical clustering with it, then visualize the clustering result:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mol.hc =<span class="st"> </span><span class="kw">hclust</span>(<span class="kw">as.dist</span>(<span class="dv">1</span> <span class="op">-</span><span class="st"> </span>simmat), <span class="dt">method =</span> <span class="st">"ward"</span>)

<span class="kw">library</span>(<span class="st">"ape"</span>)  <span class="co"># tree visualization of clusters</span>
clus5 =<span class="st"> </span><span class="kw">cutree</span>(mol.hc, <span class="dv">5</span>)  <span class="co"># cut dendrogram into 5 clusters</span>

<span class="co"># generate colors</span>
<span class="kw">library</span>(<span class="st">"RColorBrewer"</span>)
pal5 =<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/RColorBrewer/topics/ColorBrewer">brewer.pal</a></span>(<span class="dv">5</span>, <span class="st">"Set1"</span>)
<span class="kw">plot</span>(<span class="kw">as.phylo</span>(mol.hc), <span class="dt">type =</span> <span class="st">"fan"</span>,
     <span class="dt">tip.color =</span> pal5[clus5],
     <span class="dt">label.offset =</span> <span class="fl">0.1</span>, <span class="dt">cex =</span> <span class="fl">0.7</span>)</code></pre></div>
<p>The clustering result for these molecules is shown in Figure 5.</p>
<img src="figures/ex4-1.png" width="65%" class="img-responsive" align="center" alt="Tree visualization of the molecular clustering result"><p align="center">
Figure 5: Tree visualization of the molecular clustering result
</p>
</div>
<div id="structure-based-chemical-similarity-searching" class="section level2">
<h2 class="hasAnchor">
<a href="#structure-based-chemical-similarity-searching" class="anchor"></a>Structure-Based Chemical Similarity Searching</h2>
<p>Structure-based chemical similarity searching ranks molecules in a database by their similarity degree to one query molecule structure. The numerical similarity value is usually computed based on the molecular fingerprints with selected metrics or by maximum common structure search. It is one of the core techniques for ligand-based virtual screening in drug discovery.</p>
<p>The SDF file <code>DB00530.sdf</code> below is retrieved from DrugBank, the drug ID DB00530 is <em>Erlotinib</em>, which is a reversible tyrosine kinase inhibitor. Given this molecule as the query molecule, we will do a similarity searching in the database <code>tyrphostin.sdf</code> presented in the last subsection.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"Rcpi"</span>)

mol   =<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"compseq/DB00530.sdf"</span>, <span class="dt">package =</span> <span class="st">"Rcpi"</span>)
moldb =<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"compseq/tyrphostin.sdf"</span>, <span class="dt">package =</span> <span class="st">"Rcpi"</span>)</code></pre></div>
<p>We can do parallelized drug molecular similarity search with the <code><a href="../reference/searchDrug.html">searchDrug()</a></code> function in Rcpi. Here we choose the search criterion to be MACCS keys with cosine similarity, FP2 fingerprints with tanimoto similarity, and maximum common substructure search with tanimoto similarity.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">rank1 =<span class="st"> </span><span class="kw"><a href="../reference/searchDrug.html">searchDrug</a></span>(
  mol, moldb, <span class="dt">cores =</span> <span class="dv">4</span>, <span class="dt">method =</span> <span class="st">"fp"</span>,
  <span class="dt">fptype =</span> <span class="st">"maccs"</span>, <span class="dt">fpsim =</span> <span class="st">"tanimoto"</span>)
rank2 =<span class="st"> </span><span class="kw"><a href="../reference/searchDrug.html">searchDrug</a></span>(
  mol, moldb, <span class="dt">cores =</span> <span class="dv">4</span>, <span class="dt">method =</span> <span class="st">"fp"</span>,
  <span class="dt">fptype =</span> <span class="st">"fp2"</span>, <span class="dt">fpsim =</span> <span class="st">"cosine"</span>)
rank3 =<span class="st"> </span><span class="kw"><a href="../reference/searchDrug.html">searchDrug</a></span>(
  mol, moldb, <span class="dt">cores =</span> <span class="dv">4</span>, <span class="dt">method =</span> <span class="st">"mcs"</span>,
  <span class="dt">mcssim =</span> <span class="st">"tanimoto"</span>)</code></pre></div>
<p>The returned search result is stored as a numerical vector, the name of each element is the molecule number in the database, and the value is the similarity value between the query molecule and this molecule. We print the top search results here:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(rank1)
##        92       100        83       101         1        36
## 0.6491228 0.6491228 0.5882353 0.5660377 0.5000000 0.4861111

<span class="kw">head</span>(rank2)
##       100        92        83       101        94        16
## 0.8310005 0.8208663 0.5405856 0.5033150 0.4390790 0.4274081

<span class="kw">head</span>(rank3)
##        92       100        23        39        94        64
## 0.7000000 0.7000000 0.4000000 0.4000000 0.4000000 0.3783784</code></pre></div>
<p>The Rcpi package also integrated the functionality of converting molecular file formats. For example, we can convert the SDF files to SMILES files using <code><a href="../reference/convMolFormat.html">convMolFormat()</a></code>. Since the #92 molecule ranks the highest in the three searches performed, we will calculate the similarity derived by maximum common substructure search between the query molecule and the #92 molecule using <code><a href="../reference/calcDrugMCSSim.html">calcDrugMCSSim()</a></code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># convert SDF format to SMILES format</span>
<span class="kw"><a href="../reference/convMolFormat.html">convMolFormat</a></span>(
    <span class="dt">infile =</span> mol, <span class="dt">outfile =</span> <span class="st">"DB00530.smi"</span>, <span class="dt">from =</span> <span class="st">"sdf"</span>, <span class="dt">to =</span> <span class="st">"smiles"</span>)
<span class="kw"><a href="../reference/convMolFormat.html">convMolFormat</a></span>(
    <span class="dt">infile =</span> moldb, <span class="dt">outfile =</span> <span class="st">"tyrphostin.smi"</span>, <span class="dt">from =</span> <span class="st">"sdf"</span>, <span class="dt">to =</span> <span class="st">"smiles"</span>)

smi1 =<span class="st"> </span><span class="kw">readLines</span>(<span class="st">"DB00530.smi"</span>)
smi2 =<span class="st"> </span><span class="kw">readLines</span>(<span class="st">"tyrphostin.smi"</span>)[<span class="dv">92</span>]  <span class="co"># select the #92 molecule</span>
<span class="kw"><a href="../reference/calcDrugMCSSim.html">calcDrugMCSSim</a></span>(smi1, smi2, <span class="dt">type =</span> <span class="st">"smile"</span>, <span class="dt">plot =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p>The MCS search result is stored in a list, which contains the original MCS result provided by the <code>fmcsR</code> package <span class="citation">Wang <em>et al.</em> (2013)</span>, the Tanimoto coefficient and the overlap coefficient.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## [[1]]
## An instance of "MCS"
##  Number of MCSs: 1
##  530: 29 atoms
##  4705: 22 atoms
##  MCS: 18 atoms
##  Tanimoto Coefficient: 0.54545
##  Overlap Coefficient: 0.81818
##
## [[2]]
## Tanimoto_Coefficient
##            0.5454545
##
## [[3]]
## Overlap_Coefficient
##           0.8181818</code></pre></div>
<p>By using <code><a href="../reference/calcDrugMCSSim.html">calcDrugMCSSim(..., plot = TRUE)</a></code>, the maximum common substructure of the two molecules is presented in Figure 6.</p>
<img src="figures/ex5-1.png" width="50%" class="img-responsive" align="center"><p align="center">
Figure 6: Maximum common structure of the query molecule and the #92 molecule in the chemical database
</p>
</div>
</div>
<div id="applications-in-chemogenomics" class="section level1">
<h1 class="hasAnchor">
<a href="#applications-in-chemogenomics" class="anchor"></a>Applications in Chemogenomics</h1>
<p>For chemogenomics modeling, Rcpi calculates compound-protein interaction (CPI) descriptors and protein-protein interaction (PPI) descriptors.</p>
<div id="predicting-drug-target-interaction-by-integrating-chemical-and-genomic-spaces" class="section level2">
<h2 class="hasAnchor">
<a href="#predicting-drug-target-interaction-by-integrating-chemical-and-genomic-spaces" class="anchor"></a>Predicting Drug-Target Interaction by Integrating Chemical and Genomic Spaces</h2>
<p>The prediction of novel interactions between drugs and target proteins is a key area in genomic drug discovery. In this example, we use the G Protein-Coupled Receptor (GPCR) dataset provided by <span class="citation">Yamanishi <em>et al.</em> (2008)</span> as our benchmark dataset.</p>
<p>A drug-target interaction network can be naturally modeled as a bipartite graph, where the nodes are target proteins or drug molecules and edges (only drugs and proteins could be connected by edges) represent drug-target interactions. Initially, the graph only contains edges describing the <em>real</em> drug-target interactions determined by experiments or other ways. In this example, all real drug-target interaction pairs (i.e., 635 drug-target interactions) are used as the positive samples. For negative samples we select random, non-interacting pairs from these drugs and proteins. They are constructed as follows:</p>
<ol style="list-style-type: decimal">
<li>Separate the pairs in the above positive samples into single drugs and proteins;</li>
<li>Re-couple these singles into pairs in a way that none of them occurs in the corresponding positive dataset.</li>
</ol>
<p>Ten generated negative sets were used in <span class="citation">Dong-Sheng Cao <em>et al.</em> (2012)</span>, here we only used one of them for a demonstration. The drug ID and target ID is stored in <code>GPCR.csv</code>. The first column is KEGG protein ID, and the second column is KEGG drug ID. The first 635 rows form the positive set, and the last 635 rows form the negative set.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"Rcpi"</span>)

gpcr =<span class="st"> </span><span class="kw">read.table</span>(<span class="kw">system.file</span>(
    <span class="st">"vignettedata/GPCR.csv"</span>, <span class="dt">package =</span> <span class="st">"Rcpi"</span>),
    <span class="dt">header =</span> <span class="ot">FALSE</span>, <span class="dt">as.is =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p>Take a look at the data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(gpcr)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">##          V1     V2
## 1 hsa:10161 D00528
## 2 hsa:10800 D00411
## 3 hsa:10800 D01828
## 4 hsa:10800 D05129
## 5 hsa:11255 D00234
## 6 hsa:11255 D00300</code></pre></div>
<p>We will visualize the network first. Figure 7 shows the connection pattern for the GPCR drug-target interaction network in the form of an <em>arc diagram</em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"igraph"</span>)
<span class="kw">library</span>(<span class="st">"arcdiagram"</span>)
<span class="kw">library</span>(<span class="st">"reshape"</span>)

g =<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/igraph/topics/graph_from_data_frame">graph.data.frame</a></span>(gpcr[<span class="dv">1</span><span class="op">:</span>(<span class="kw">nrow</span>(gpcr)<span class="op">/</span><span class="dv">2</span>), ], <span class="dt">directed =</span> <span class="ot">FALSE</span>)
edgelist =<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/igraph/topics/as_edgelist">get.edgelist</a></span>(g)
vlabels  =<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/igraph/topics/V">V</a></span>(g)<span class="op">$</span>name
vgroups  =<span class="st"> </span><span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>, <span class="dv">95</span>), <span class="kw">rep</span>(<span class="dv">1</span>, <span class="dv">223</span>))
vfill    =<span class="st"> </span><span class="kw">c</span>(<span class="kw">rep</span>(<span class="st">"#8B91D4"</span>, <span class="dv">95</span>), <span class="kw">rep</span>(<span class="st">"#B2C771"</span>, <span class="dv">223</span>))
vborders =<span class="st"> </span><span class="kw">c</span>(<span class="kw">rep</span>(<span class="st">"#6F74A9"</span>, <span class="dv">95</span>), <span class="kw">rep</span>(<span class="st">"#8E9F5A"</span>, <span class="dv">223</span>))
degrees  =<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/igraph/topics/degree">degree</a></span>(g)

xx =<span class="st"> </span><span class="kw">data.frame</span>(vgroups, degrees, vlabels, <span class="dt">ind =</span> <span class="dv">1</span><span class="op">:</span><span class="kw"><a href="http://www.rdocumentation.org/packages/igraph/topics/gorder">vcount</a></span>(g))
yy =<span class="st"> </span><span class="kw">arrange</span>(xx, <span class="kw">desc</span>(vgroups), <span class="kw">desc</span>(degrees))
new.ord =<span class="st"> </span>yy<span class="op">$</span>ind

<span class="kw">arcplot</span>(
  edgelist, <span class="dt">ordering =</span> new.ord, <span class="dt">labels =</span> vlabels,
  <span class="dt">cex.labels =</span> <span class="fl">0.1</span>, <span class="dt">show.nodes =</span> <span class="ot">TRUE</span>,
  <span class="dt">col.nodes =</span> vborders, <span class="dt">bg.nodes =</span> vfill,
  <span class="dt">cex.nodes =</span> <span class="kw">log10</span>(degrees) <span class="op">+</span><span class="st"> </span><span class="fl">0.1</span>,
  <span class="dt">pch.nodes =</span> <span class="dv">21</span>, <span class="dt">line =</span> <span class="op">-</span><span class="fl">0.5</span>, <span class="dt">col.arcs =</span> <span class="kw">hsv</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">0.2</span>, <span class="fl">0.25</span>))</code></pre></div>
<img src="figures/ex6-1.png" width="85%" class="img-responsive" align="center" alt="Arc diagram visualization of the GPCR drug-target interaction network"><p align="center">
Figure 7: Arc diagram visualization of the GPCR drug-target interaction network
</p>
<p>An arc diagram visualize the nodes in the network in a one-dimensional layout, while using circular arcs to represent edges. With a good ordering of nodes, it is easy to identify cliques and bridges.</p>
<p>Next, we will download the target protein sequences (in FASTA format) and drug molecule (in SMILES format) from the KEGG database, in parallel:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"Rcpi"</span>)

gpcr =<span class="st"> </span><span class="kw">read.table</span>(<span class="kw">system.file</span>(
  <span class="st">"vignettedata/GPCR.csv"</span>, <span class="dt">package =</span> <span class="st">"Rcpi"</span>),
  <span class="dt">header =</span> <span class="ot">FALSE</span>, <span class="dt">as.is =</span> <span class="ot">TRUE</span>)

protid =<span class="st"> </span><span class="kw">unique</span>(gpcr[, <span class="dv">1</span>])
drugid =<span class="st"> </span><span class="kw">unique</span>(gpcr[, <span class="dv">2</span>])

protseq =<span class="st"> </span><span class="kw"><a href="../reference/getSeqFromKEGG.html">getSeqFromKEGG</a></span>(protid, <span class="dt">parallel =</span> <span class="dv">5</span>)
drugseq =<span class="st"> </span><span class="kw"><a href="../reference/getSmiFromKEGG.html">getSmiFromKEGG</a></span>(drugid, <span class="dt">parallel =</span> <span class="dv">50</span>)</code></pre></div>
<p>If the connection is slow or accidentally interrupts, just try a few more times until success.</p>
<p>The functions in Rcpi named after <code>getMolFrom...()</code> and <code>getSmiFrom...()</code> supports the parallelized retrieval of (drug) molecules from PubChem, ChEMBL, CAS, KEGG, and DrugBank. The functions named after <code>getSeqFrom...()</code>, <code>getFASTAFrom...()</code>, and <code>getPDBFrom...()</code> supports the parallelized retrieval of proteins from UniProt, KEGG and RCSB PDB. The functions <code><a href="../reference/getDrug.html">getDrug()</a></code> and <code><a href="../reference/getProt.html">getProt()</a></code> are two integrated wrapper functions for downloading the molecules and protein sequences from these online databases.</p>
<p>After the sequences were downloaded, we can calculate the protein sequence descriptors and molecular descriptors for the targets and drugs:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">x0.prot =<span class="st"> </span><span class="kw">cbind</span>(
  <span class="kw">t</span>(<span class="kw">sapply</span>(<span class="kw">unlist</span>(protseq), extractProtAPAAC)),
  <span class="kw">t</span>(<span class="kw">sapply</span>(<span class="kw">unlist</span>(protseq), extractProtCTriad)))

x0.drug =<span class="st"> </span><span class="kw">cbind</span>(
  <span class="kw"><a href="../reference/extractDrugEstateComplete.html">extractDrugEstateComplete</a></span>(<span class="kw"><a href="../reference/readMolFromSmi.html">readMolFromSmi</a></span>(<span class="kw">textConnection</span>(drugseq))),
  <span class="kw"><a href="../reference/extractDrugMACCSComplete.html">extractDrugMACCSComplete</a></span>(<span class="kw"><a href="../reference/readMolFromSmi.html">readMolFromSmi</a></span>(<span class="kw">textConnection</span>(drugseq))),
  <span class="kw"><a href="../reference/extractDrugOBFP4.html">extractDrugOBFP4</a></span>(drugseq, <span class="dt">type =</span> <span class="st">"smile"</span>))</code></pre></div>
<p>Since the descriptors is only for the <em>deduplicated</em> drug and target list, we need to generate the full descriptor matrix for the training data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># generate drug x / protein x / y</span>
x.prot =<span class="st"> </span><span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">nrow =</span> <span class="kw">nrow</span>(gpcr), <span class="dt">ncol =</span> <span class="kw">ncol</span>(x0.prot))
x.drug =<span class="st"> </span><span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">nrow =</span> <span class="kw">nrow</span>(gpcr), <span class="dt">ncol =</span> <span class="kw">ncol</span>(x0.drug))
<span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(gpcr)) x.prot[i, ] =<span class="st"> </span>x0.prot[<span class="kw">which</span>(gpcr[, <span class="dv">1</span>][i] <span class="op">==</span><span class="st"> </span>protid), ]
<span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(gpcr)) x.drug[i, ] =<span class="st"> </span>x0.drug[<span class="kw">which</span>(gpcr[, <span class="dv">2</span>][i] <span class="op">==</span><span class="st"> </span>drugid), ]

y =<span class="st"> </span><span class="kw">as.factor</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="st">"pos"</span>, <span class="kw">nrow</span>(gpcr)<span class="op">/</span><span class="dv">2</span>), <span class="kw">rep</span>(<span class="st">"neg"</span>, <span class="kw">nrow</span>(gpcr)<span class="op">/</span><span class="dv">2</span>)))</code></pre></div>
<p>Generate drug-target interaction descriptors using <code><a href="../reference/getCPI.html">getCPI()</a></code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">x =<span class="st"> </span><span class="kw"><a href="../reference/getCPI.html">getCPI</a></span>(x.prot, x.drug, <span class="dt">type =</span> <span class="st">"combine"</span>)</code></pre></div>
<p>The pairwise interaction is another useful type of representation in drug-target prediction, protein-protein interaction prediction and related research. Rcpi also provides <code><a href="../reference/getPPI.html">getPPI()</a></code> to generate protein-protein interaction descriptors. <code><a href="../reference/getPPI.html">getPPI()</a></code> provides three types of interactions while <code><a href="../reference/getCPI.html">getCPI()</a></code> provides two types. The argument <code>type</code> is used to control this.</p>
<div id="compound-protein-interaction-cpi-descriptors" class="section level3">
<h3 class="hasAnchor">
<a href="#compound-protein-interaction-cpi-descriptors" class="anchor"></a>Compound-protein interaction (CPI) descriptors</h3>
<p>For compound descriptor vector <span class="math inline">\(d_1^{1 \times p_1}\)</span> and the protein descriptor vector <span class="math inline">\(d_2^{1 \times p_2}\)</span>, there are two methods for construction of descriptor vector <span class="math inline">\(d\)</span> for compound-protein interaction:</p>
<ol style="list-style-type: decimal">
<li>
<code>type = "combine"</code> - combine the two feature matrix, <span class="math inline">\(d\)</span> has <span class="math inline">\(p_1 + p_2\)</span> columns;</li>
<li>
<code>type = "tensorprod"</code> - column-by-column (pseudo)-tensor product type interactions, <span class="math inline">\(d\)</span> has <span class="math inline">\(p_1 \times p_2\)</span> columns.</li>
</ol>
</div>
<div id="protein-protein-interaction-ppi-descriptors" class="section level3">
<h3 class="hasAnchor">
<a href="#protein-protein-interaction-ppi-descriptors" class="anchor"></a>Protein-protein interaction (PPI) descriptors</h3>
<p>For interaction protein A and protein B, let <span class="math inline">\(d_1^{1 \times p}\)</span> and <span class="math inline">\(d_2^{1 \times p}\)</span> be the descriptor vectors. There are three methods to construct the protein-protein interaction descriptor <span class="math inline">\(d\)</span>:</p>
<ol style="list-style-type: decimal">
<li>
<code>type = "combine"</code> - combine the two descriptor matrix, <span class="math inline">\(d\)</span> has <span class="math inline">\(p + p\)</span> columns;</li>
<li>
<code>type = "tensorprod"</code> - column-by-column (pseudo)-tensor product type interactions, <span class="math inline">\(d\)</span> has <span class="math inline">\(p \times p\)</span> columns;</li>
<li>
<code>type = "entrywise"</code> - entrywise product and entrywise sum of the two matrices, then combine them, <span class="math inline">\(d\)</span> has <span class="math inline">\(p + p\)</span> columns.</li>
</ol>
<p>Train a random forest classification model with 5-fold repeated CV:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"caret"</span>)
x =<span class="st"> </span>x[, <span class="op">-</span><span class="kw">nearZeroVar</span>(x)]

<span class="co"># cross-validation settings</span>
ctrl =<span class="st"> </span><span class="kw">trainControl</span>(
    <span class="dt">method =</span> <span class="st">"cv"</span>, <span class="dt">number =</span> <span class="dv">5</span>, <span class="dt">repeats =</span> <span class="dv">10</span>,
    <span class="dt">classProbs =</span> <span class="ot">TRUE</span>,
    <span class="dt">summaryFunction =</span> twoClassSummary)

<span class="co"># train a random forest classifier</span>
<span class="kw">library</span>(<span class="st">"randomForest"</span>)

<span class="kw">set.seed</span>(<span class="dv">1006</span>)
rf.fit =<span class="st"> </span><span class="kw">train</span>(
  x, y, <span class="dt">method =</span> <span class="st">"rf"</span>, <span class="dt">trControl =</span> ctrl,
  <span class="dt">metric =</span> <span class="st">"ROC"</span>, <span class="dt">preProc =</span> <span class="kw">c</span>(<span class="st">"center"</span>, <span class="st">"scale"</span>))
\end{CodeInput}

Print the cross<span class="op">-</span>validation result<span class="op">:</span>

\begin{CodeInput}
<span class="kw">print</span>(rf.fit)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Random Forest
##
## 1270 samples
##  562 predictors
##    2 classes: "neg", "pos"
##
## Pre-processing: centered, scaled
## Resampling: Cross-Validated (5 fold)
##
## Summary of sample sizes: 1016, 1016, 1016, 1016, 1016
##
## Resampling results across tuning parameters:
##
##   mtry  ROC    Sens   Spec   ROC SD  Sens SD  Spec SD
##   2     0.83   0.726  0.778  0.0221  0.044    0.0395
##   33    0.882  0.795  0.82   0.018   0.0522   0.0443
##   562   0.893  0.822  0.844  0.0161  0.0437   0.0286
##
## ROC was used to select the optimal model using  the largest value.
## The final value used for the model was mtry = 562.</code></pre></div>
<p>Predict on the training set (for demonstration purpose only) and plot ROC curve.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">rf.pred =<span class="st"> </span><span class="kw">predict</span>(rf.fit<span class="op">$</span>finalModel, x, <span class="dt">type =</span> <span class="st">"prob"</span>)[, <span class="dv">1</span>]

<span class="kw">library</span>(<span class="st">"pROC"</span>)
<span class="kw">plot</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/pROC/topics/smooth">smooth</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/pROC/topics/roc">roc</a></span>(y, rf.pred)), <span class="dt">grid =</span> <span class="ot">TRUE</span>, <span class="dt">print.auc =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p>The ROC curve is shown in Figure 8.</p>
<img src="figures/ex6-2.png" width="65%" class="img-responsive" align="center" alt="ROC curve for predicting on the training set of the GPCR drug-target interaction dataset using random forest"><p align="center">
Figure 8: ROC curve for predicting on the training set of the GPCR drug-target interaction dataset using random forest
</p>
</div>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-cao2012large">
<p>Cao,D.-S. <em>et al.</em> (2012) Large-scale prediction of drug-target interactions using protein sequences and drug topological structures. <em>Analytica chimica acta</em>, <strong>752</strong>, 1â€“10.</p>
</div>
<div id="ref-Rcpi2015">
<p>Cao,D.-S. <em>et al.</em> (2015) Rcpi: R/Bioconductor package to generate various descriptors of proteins, compounds and their interactions. <em>Bioinformatics</em>, <strong>31</strong>, 279â€“281.</p>
</div>
<div id="ref-cao2012silico">
<p>Cao,D.-S. <em>et al.</em> (2012) In silico toxicity prediction by support vector machine and smiles representation-based string kernel. <em>SAR and QSAR in Environmental Research</em>, <strong>23</strong>, 141â€“153.</p>
</div>
<div id="ref-chouapaac">
<p>Chou,K.-C. (2005) Using amphiphilic pseudo amino acid composition to predict enzyme subfamily classes. <em>Bioinformatics</em>, <strong>21</strong>, 10â€“19.</p>
</div>
<div id="ref-chou2008cell">
<p>Chou,K.-C. and Shen,H.-B. (2008) Cell-ploc: A package of web servers for predicting subcellular localization of proteins in various organisms. <em>Nature Protocols</em>, <strong>3</strong>, 153â€“162.</p>
</div>
<div id="ref-chemmineob">
<p>Horan,K. and Girke,T. (2013) ChemmineOB: R interface to a subset of openbabel functionalities.</p>
</div>
<div id="ref-steinbeck2003chemistry">
<p>Steinbeck,C. <em>et al.</em> (2003) The chemistry development kit (cdk): An open-source java library for chemo-and bioinformatics. <em>Journal of chemical information and computer sciences</em>, <strong>43</strong>, 493â€“500.</p>
</div>
<div id="ref-wang2013fmcsr">
<p>Wang,Y. <em>et al.</em> (2013) FmcsR: Mismatch tolerant maximum common substructure searching in r. <em>Bioinformatics</em>, <strong>29</strong>, 2792â€“2794.</p>
</div>
<div id="ref-yamanishi2008prediction">
<p>Yamanishi,Y. <em>et al.</em> (2008) Prediction of drugâ€“target interaction networks from the integration of chemical and genomic spaces. <em>Bioinformatics</em>, <strong>24</strong>, i232â€“i240.</p>
</div>
<div id="ref-yan2012comparison">
<p>Yan,J. <em>et al.</em> (2012) Comparison of quantitative structureâ€“retention relationship models on four stationary phases with different polarity for a diverse set of flavor compounds. <em>Journal of Chromatography A</em>, <strong>1223</strong>, 118â€“125.</p>
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#installation">Installation</a></li>
      <li><a href="#how-to-cite-rcpi">How to Cite Rcpi</a></li>
      </ul>
</li>
      <li>
<a href="#applications-in-bioinformatics">Applications in Bioinformatics</a><ul class="nav nav-pills nav-stacked">
<li><a href="#predicting-protein-subcellular-localization">Predicting Protein Subcellular Localization</a></li>
      </ul>
</li>
      <li>
<a href="#applications-in-chemoinformatics">Applications in Chemoinformatics</a><ul class="nav nav-pills nav-stacked">
<li><a href="#regression-modeling-in-qsrr-study-of-retention-indices">Regression Modeling in QSRR Study of Retention Indices</a></li>
      <li><a href="#in-silico-toxicity-classification-for-drug-discovery"><em>In Silico</em> Toxicity Classification for Drug Discovery</a></li>
      <li><a href="#clustering-of-molecules-based-on-structural-similarities">Clustering of Molecules Based on Structural Similarities</a></li>
      <li><a href="#structure-based-chemical-similarity-searching">Structure-Based Chemical Similarity Searching</a></li>
      </ul>
</li>
      <li>
<a href="#applications-in-chemogenomics">Applications in Chemogenomics</a><ul class="nav nav-pills nav-stacked">
<li><a href="#predicting-drug-target-interaction-by-integrating-chemical-and-genomic-spaces">Predicting Drug-Target Interaction by Integrating Chemical and Genomic Spaces</a></li>
      </ul>
</li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Nan Xiao, Dong-Sheng Cao, Qing-Song Xu.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
