<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Rcpi Quick Reference Card â€¢ Rcpi</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/spacelab/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Rcpi</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/index.html">Articles</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/road2stat/Rcpi">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>Rcpi Quick Reference Card</h1>
                        <h4 class="author">Nan Xiao &lt;<a href="https://nanx.me" class="uri">https://nanx.me</a>&gt;</h4>
            
            <h4 class="date">2017-06-09</h4>
          </div>

    
    
<div class="contents">
<div id="retrieve-protein-sequence-data-from-online-databases" class="section level1">
<h1 class="hasAnchor">
<a href="#retrieve-protein-sequence-data-from-online-databases" class="anchor"></a>Retrieve Protein Sequence Data from Online Databases</h1>
<table class="table">
<caption>Table 1: Retrieving protein sequence data from various online databases</caption>
<thead><tr class="header">
<th align="left">Function name</th>
<th align="left">Function description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">getProt()</td>
<td align="left">Retrieve protein sequence in FASTA format or PDB format from various online databases</td>
</tr>
<tr class="even">
<td align="left">getFASTAFromUniProt()</td>
<td align="left">Retrieve protein sequence in FASTA format from UniProt</td>
</tr>
<tr class="odd">
<td align="left">getFASTAFromKEGG()</td>
<td align="left">Retrieve protein sequence in FASTA format from KEGG</td>
</tr>
<tr class="even">
<td align="left">getPDBFromRCSBPDB()</td>
<td align="left">Retrieve protein sequence in PDB Format from RCSB PDB</td>
</tr>
<tr class="odd">
<td align="left">getSeqFromUniProt()</td>
<td align="left">Retrieve protein sequence from UniProt</td>
</tr>
<tr class="even">
<td align="left">getSeqFromKEGG()</td>
<td align="left">Retrieve protein sequence from KEGG</td>
</tr>
<tr class="odd">
<td align="left">getSeqFromRCSBPDB()</td>
<td align="left">Retrieve protein sequence from RCSB PDB</td>
</tr>
</tbody>
</table>
</div>
<div id="retrieve-drug-molecular-data-from-online-databases" class="section level1">
<h1 class="hasAnchor">
<a href="#retrieve-drug-molecular-data-from-online-databases" class="anchor"></a>Retrieve Drug Molecular Data from Online Databases</h1>
<table class="table">
<caption>Table 2: Retrieving drug molecular data from various online databases</caption>
<thead><tr class="header">
<th align="left">Function name</th>
<th align="left">Function description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">getDrug()</td>
<td align="left">Retrieve drug molecules in MOL format and SMILES format from various online databases</td>
</tr>
<tr class="even">
<td align="left">getMolFromDrugBank()</td>
<td align="left">Retrieve drug molecules in MOL format from DrugBank</td>
</tr>
<tr class="odd">
<td align="left">getMolFromPubChem()</td>
<td align="left">Retrieve drug molecules in MOL format from PubChem</td>
</tr>
<tr class="even">
<td align="left">getMolFromChEMBL()</td>
<td align="left">Retrieve drug molecules in MOL format from ChEMBL</td>
</tr>
<tr class="odd">
<td align="left">getMolFromKEGG()</td>
<td align="left">Retrieve drug molecules in MOL format from the KEGG</td>
</tr>
<tr class="even">
<td align="left">getMolFromCAS()</td>
<td align="left">Retrieve drug molecules in InChI format from CAS</td>
</tr>
<tr class="odd">
<td align="left">getSmiFromDrugBank()</td>
<td align="left">Retrieve drug molecules in SMILES format from DrugBank</td>
</tr>
<tr class="even">
<td align="left">getSmiFromPubChem()</td>
<td align="left">Retrieve drug molecules in SMILES format from PubChem</td>
</tr>
<tr class="odd">
<td align="left">getSmiFromChEMBL()</td>
<td align="left">Retrieve drug molecules in SMILES format from ChEMBL</td>
</tr>
<tr class="even">
<td align="left">getSmiFromKEGG()</td>
<td align="left">Retrieve drug molecules in SMILES format from KEGG</td>
</tr>
</tbody>
</table>
</div>
<div id="calculate-commonly-used-protein-sequence-derived-descriptors" class="section level1">
<h1 class="hasAnchor">
<a href="#calculate-commonly-used-protein-sequence-derived-descriptors" class="anchor"></a>Calculate Commonly Used Protein Sequence Derived Descriptors</h1>
<table class="table">
<caption>Table 3: Calculating commonly used protein sequence derived descriptors</caption>
<thead><tr class="header">
<th align="left">Function name</th>
<th align="left">Descriptor name</th>
<th align="left">Descriptor group</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">extractProtAAC()</td>
<td align="left">Amino acid composition</td>
<td align="left">Amino acid composition</td>
</tr>
<tr class="even">
<td align="left">extractProtDC()</td>
<td align="left">Dipeptide composition</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">extractProtTC()</td>
<td align="left">Tripeptide composition</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">extractProtMoreauBroto()</td>
<td align="left">Normalized Moreau-Broto autocorrelation</td>
<td align="left">Autocorrelation</td>
</tr>
<tr class="odd">
<td align="left">extractProtMoran()</td>
<td align="left">Moran autocorrelation</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">extractProtGeary()</td>
<td align="left">Geary autocorrelation</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">extractProtCTDC()</td>
<td align="left">Composition</td>
<td align="left">CTD</td>
</tr>
<tr class="even">
<td align="left">extractProtCTDT()</td>
<td align="left">Transition</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">extractProtCTDD()</td>
<td align="left">Distribution</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">extractProtCTriad()</td>
<td align="left">Conjoint Triad</td>
<td align="left">Conjoint Triad</td>
</tr>
<tr class="odd">
<td align="left">extractProtSOCN()</td>
<td align="left">Sequence-order-coupling number</td>
<td align="left">Quasi-sequence-order</td>
</tr>
<tr class="even">
<td align="left">extractProtQSO()</td>
<td align="left">Quasi-sequence-order descriptors</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">extractProtPAAC()</td>
<td align="left">Pseudo-amino acid composition</td>
<td align="left">Pseudo-amino acid composition</td>
</tr>
<tr class="even">
<td align="left">extractProtAPAAC()</td>
<td align="left">Amphiphilic pseudo-amino acid composition</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">AAindex</td>
<td align="left">AAindex data of 544 physicochemical and biological properties for 20 amino acids</td>
<td align="left">Dataset</td>
</tr>
</tbody>
</table>
</div>
<div id="generate-profile-based-protein-representations" class="section level1">
<h1 class="hasAnchor">
<a href="#generate-profile-based-protein-representations" class="anchor"></a>Generate Profile-Based Protein Representations</h1>
<table class="table">
<caption>Table 4: Generating profile-based protein representations</caption>
<thead><tr class="header">
<th align="left">Function name</th>
<th align="left">Function description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">extractProtPSSM()</td>
<td align="left">Compute PSSM (Position-Specific Scoring Matrix) for given protein sequence or peptides</td>
</tr>
<tr class="even">
<td align="left">extractProtPSSMFeature()</td>
<td align="left">Profile-based protein representation derived by PSSM</td>
</tr>
<tr class="odd">
<td align="left">extractProtPSSMAcc()</td>
<td align="left">Profile-based protein representation derived by PSSM and auto cross covariance (ACC)</td>
</tr>
</tbody>
</table>
</div>
<div id="generate-scales-based-descriptors-for-proteochemometrics-modeling" class="section level1">
<h1 class="hasAnchor">
<a href="#generate-scales-based-descriptors-for-proteochemometrics-modeling" class="anchor"></a>Generate Scales-Based Descriptors for Proteochemometrics Modeling</h1>
<table class="table">
<caption>Table 5: Generating scales-based descriptors for proteochemometrics modeling</caption>
<thead><tr class="header">
<th align="left">Function name</th>
<th align="left">Descriptor class</th>
<th align="left">Derived by</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">extractPCMScales()</td>
<td align="left">Generalized scales-based descriptors derived by principal components analysis (PCA)</td>
<td align="left">Principal components analysis</td>
</tr>
<tr class="even">
<td align="left">extractPCMPropScales()</td>
<td align="left">Generalized scales-based descriptors derived by amino acid properties (AAindex)</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">extractPCMDescScales()</td>
<td align="left">Generalized scales-based descriptors derived by 2D and 3D molecular descriptors (Topological, WHIM, VHSE, etc.)</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">extractPCMFAScales()</td>
<td align="left">Generalized scales-based descriptors derived by factor analysis</td>
<td align="left">Factor analysis</td>
</tr>
<tr class="odd">
<td align="left">extractPCMMDSScales()</td>
<td align="left">Generalized scales-based descriptors derived by multidimensional scaling (MDS)</td>
<td align="left">Multidimensional scaling</td>
</tr>
<tr class="even">
<td align="left">extractPCMBLOSUM()</td>
<td align="left">Generalized BLOSUM and PAM matrix-derived descriptors</td>
<td align="left">Substitution matrix</td>
</tr>
<tr class="odd">
<td align="left">acc()</td>
<td align="left">Auto cross covariance (ACC) for generating scales-based descriptors of the same length</td>
<td align="left"></td>
</tr>
</tbody>
</table>
</div>
<div id="molecular-descriptor-sets-of-the-20-amino-acids-for-generating-scales-based-descriptors" class="section level1">
<h1 class="hasAnchor">
<a href="#molecular-descriptor-sets-of-the-20-amino-acids-for-generating-scales-based-descriptors" class="anchor"></a>Molecular Descriptor Sets of the 20 Amino Acids for Generating Scales-Based Descriptors</h1>
<table class="table">
<caption>Table 6: Pre-calculated molecular descriptor sets of the 20 amino acids in for generating scales-based descriptors for proteochemometrics modeling.</caption>
<thead><tr class="header">
<th align="left">Dataset name</th>
<th align="left">Dataset description</th>
<th align="left">Dimensionality</th>
<th align="left">Calculated by</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">OptAA3d</td>
<td align="left">Optimized 20 amino acids</td>
<td align="left">â€“</td>
<td align="left">MOE</td>
</tr>
<tr class="even">
<td align="left">AA2DACOR</td>
<td align="left">2D autocorrelations descriptors</td>
<td align="left">92</td>
<td align="left">Dragon</td>
</tr>
<tr class="odd">
<td align="left">AA3DMoRSE</td>
<td align="left">3D-MoRSE descriptors</td>
<td align="left">160</td>
<td align="left">Dragon</td>
</tr>
<tr class="even">
<td align="left">AAACF</td>
<td align="left">Atom-centred fragments descriptors</td>
<td align="left">6</td>
<td align="left">Dragon</td>
</tr>
<tr class="odd">
<td align="left">AABurden</td>
<td align="left">Burden Eigenvalues descriptors</td>
<td align="left">62</td>
<td align="left">Dragon</td>
</tr>
<tr class="even">
<td align="left">AAConn</td>
<td align="left">Connectivity indices descriptors</td>
<td align="left">33</td>
<td align="left">Dragon</td>
</tr>
<tr class="odd">
<td align="left">AAConst</td>
<td align="left">Constitutional descriptors</td>
<td align="left">23</td>
<td align="left">Dragon</td>
</tr>
<tr class="even">
<td align="left">AAEdgeAdj</td>
<td align="left">Edge adjacency indices descriptors</td>
<td align="left">97</td>
<td align="left">Dragon</td>
</tr>
<tr class="odd">
<td align="left">AAEigIdx</td>
<td align="left">Eigenvalue-based indices descriptors</td>
<td align="left">44</td>
<td align="left">Dragon</td>
</tr>
<tr class="even">
<td align="left">AAFGC</td>
<td align="left">Functional group counts descriptors</td>
<td align="left">5</td>
<td align="left">Dragon</td>
</tr>
<tr class="odd">
<td align="left">AAGeom</td>
<td align="left">Geometrical descriptors</td>
<td align="left">41</td>
<td align="left">Dragon</td>
</tr>
<tr class="even">
<td align="left">AAGETAWAY</td>
<td align="left">GETAWAY descriptors</td>
<td align="left">194</td>
<td align="left">Dragon</td>
</tr>
<tr class="odd">
<td align="left">AAInfo</td>
<td align="left">Information indices descriptors</td>
<td align="left">47</td>
<td align="left">Dragon</td>
</tr>
<tr class="even">
<td align="left">AAMolProp</td>
<td align="left">Molecular properties descriptors</td>
<td align="left">12</td>
<td align="left">Dragon</td>
</tr>
<tr class="odd">
<td align="left">AARandic</td>
<td align="left">Randic molecular profiles descriptors</td>
<td align="left">41</td>
<td align="left">Dragon</td>
</tr>
<tr class="even">
<td align="left">AARDF</td>
<td align="left">RDF descriptors</td>
<td align="left">82</td>
<td align="left">Dragon</td>
</tr>
<tr class="odd">
<td align="left">AATopo</td>
<td align="left">Topological descriptors</td>
<td align="left">78</td>
<td align="left">Dragon</td>
</tr>
<tr class="even">
<td align="left">AATopoChg</td>
<td align="left">Topological charge indices descriptors</td>
<td align="left">15</td>
<td align="left">Dragon</td>
</tr>
<tr class="odd">
<td align="left">AAWalk</td>
<td align="left">Walk and path counts descriptors</td>
<td align="left">40</td>
<td align="left">Dragon</td>
</tr>
<tr class="even">
<td align="left">AAWHIM</td>
<td align="left">WHIM descriptors</td>
<td align="left">99</td>
<td align="left">Dragon</td>
</tr>
<tr class="odd">
<td align="left">AACPSA</td>
<td align="left">CPSA descriptors</td>
<td align="left">41</td>
<td align="left">Accelrys Discovery Studio</td>
</tr>
<tr class="even">
<td align="left">AADescAll</td>
<td align="left">All the 2D descriptors calculated by Dragon</td>
<td align="left">1171</td>
<td align="left">Dragon</td>
</tr>
<tr class="odd">
<td align="left">AAMOE2D</td>
<td align="left">All the 2D descriptors calculated by MOE</td>
<td align="left">148</td>
<td align="left">MOE</td>
</tr>
<tr class="even">
<td align="left">AAMOE3D</td>
<td align="left">All the 3D descriptors calculated by MOE</td>
<td align="left">143</td>
<td align="left">MOE</td>
</tr>
<tr class="odd">
<td align="left">AABLOSUM45</td>
<td align="left">BLOSUM45 matrix for 20 amino acids</td>
<td align="left"><span class="math inline">\(20 \times 20\)</span></td>
<td align="left">Biostrings</td>
</tr>
<tr class="even">
<td align="left">AABLOSUM50</td>
<td align="left">BLOSUM50 matrix for 20 amino acids</td>
<td align="left"><span class="math inline">\(20 \times 20\)</span></td>
<td align="left">Biostrings</td>
</tr>
<tr class="odd">
<td align="left">AABLOSUM62</td>
<td align="left">BLOSUM62 matrix for 20 amino acids</td>
<td align="left"><span class="math inline">\(20 \times 20\)</span></td>
<td align="left">Biostrings</td>
</tr>
<tr class="even">
<td align="left">AABLOSUM80</td>
<td align="left">BLOSUM80 matrix for 20 amino acids</td>
<td align="left"><span class="math inline">\(20 \times 20\)</span></td>
<td align="left">Biostrings</td>
</tr>
<tr class="odd">
<td align="left">AABLOSUM100</td>
<td align="left">BLOSUM100 matrix for 20 amino acids</td>
<td align="left"><span class="math inline">\(20 \times 20\)</span></td>
<td align="left">Biostrings</td>
</tr>
<tr class="even">
<td align="left">AAPAM30</td>
<td align="left">PAM30 matrix for 20 amino acids</td>
<td align="left"><span class="math inline">\(20 \times 20\)</span></td>
<td align="left">Biostrings</td>
</tr>
<tr class="odd">
<td align="left">AAPAM40</td>
<td align="left">PAM40 matrix for 20 amino acids</td>
<td align="left"><span class="math inline">\(20 \times 20\)</span></td>
<td align="left">Biostrings</td>
</tr>
<tr class="even">
<td align="left">AAPAM70</td>
<td align="left">PAM70 matrix for 20 amino acids</td>
<td align="left"><span class="math inline">\(20 \times 20\)</span></td>
<td align="left">Biostrings</td>
</tr>
<tr class="odd">
<td align="left">AAPAM120</td>
<td align="left">PAM120 matrix for 20 amino acids</td>
<td align="left"><span class="math inline">\(20 \times 20\)</span></td>
<td align="left">Biostrings</td>
</tr>
<tr class="even">
<td align="left">AAPAM250</td>
<td align="left">PAM250 matrix for 20 amino acids</td>
<td align="left"><span class="math inline">\(20 \times 20\)</span></td>
<td align="left">Biostrings</td>
</tr>
</tbody>
</table>
<p><strong>Note</strong>: non-informative descriptors (e.g.Â descriptors with only one value across all the 20 amino acids) in these datasets have been filtered out.</p>
</div>
<div id="molecular-descriptors" class="section level1">
<h1 class="hasAnchor">
<a href="#molecular-descriptors" class="anchor"></a>Molecular Descriptors</h1>
<table class="table">
<caption>Table 7: Molecular descriptors</caption>
<thead><tr class="header">
<th align="left">Function name</th>
<th align="left">Descriptor name</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">extractDrugAIO()</td>
<td align="left">All the molecular descriptors in the package</td>
</tr>
<tr class="even">
<td align="left">extractDrugALOGP()</td>
<td align="left">Atom additive logP and molar refractivity values descriptor</td>
</tr>
<tr class="odd">
<td align="left">extractDrugAminoAcidCount()</td>
<td align="left">Number of amino acids</td>
</tr>
<tr class="even">
<td align="left">extractDrugApol()</td>
<td align="left">Sum of the atomic polarizabilities</td>
</tr>
<tr class="odd">
<td align="left">extractDrugAromaticAtomsCount()</td>
<td align="left">Number of aromatic atoms</td>
</tr>
<tr class="even">
<td align="left">extractDrugAromaticBondsCount()</td>
<td align="left">Number of aromatic bonds</td>
</tr>
<tr class="odd">
<td align="left">extractDrugAtomCount()</td>
<td align="left">Number of atom descriptor</td>
</tr>
<tr class="even">
<td align="left">extractDrugAutocorrelationCharge()</td>
<td align="left">Moreau-Broto autocorrelation descriptors using partial charges</td>
</tr>
<tr class="odd">
<td align="left">extractDrugAutocorrelationMass()</td>
<td align="left">Moreau-Broto autocorrelation descriptors using atomic weight</td>
</tr>
<tr class="even">
<td align="left">extractDrugAutocorrelationPolarizability()</td>
<td align="left">Moreau-Broto autocorrelation descriptors using polarizability</td>
</tr>
<tr class="odd">
<td align="left">extractDrugBCUT()</td>
<td align="left">BCUT, the eigenvalue based descriptor</td>
</tr>
<tr class="even">
<td align="left">extractDrugBondCount()</td>
<td align="left">Number of bonds of a certain bond order</td>
</tr>
<tr class="odd">
<td align="left">extractDrugBPol()</td>
<td align="left">Sum of the absolute value of the difference between atomic polarizabilities of all bonded atoms in the molecule</td>
</tr>
<tr class="even">
<td align="left">extractDrugCarbonTypes()</td>
<td align="left">Topological descriptor characterizing the carbon connectivity in terms of hybridization</td>
</tr>
<tr class="odd">
<td align="left">extractDrugChiChain()</td>
<td align="left">Kier &amp; Hall Chi chain indices of orders 3, 4, 5, 6 and 7</td>
</tr>
<tr class="even">
<td align="left">extractDrugChiCluster()</td>
<td align="left">Kier &amp; Hall Chi cluster indices of orders 3, 4, 5 and 6</td>
</tr>
<tr class="odd">
<td align="left">extractDrugChiPath()</td>
<td align="left">Kier &amp; Hall Chi path indices of orders 0 to 7</td>
</tr>
<tr class="even">
<td align="left">extractDrugChiPathCluster()</td>
<td align="left">Kier &amp; Hall Chi path cluster indices of orders 4, 5 and 6</td>
</tr>
<tr class="odd">
<td align="left">extractDrugCPSA()</td>
<td align="left">Descriptors combining surface area and partial charge information</td>
</tr>
<tr class="even">
<td align="left">extractDrugDescOB()</td>
<td align="left">Molecular descriptors provided by OpenBabel</td>
</tr>
<tr class="odd">
<td align="left">extractDrugECI()</td>
<td align="left">Eccentric connectivity index descriptor</td>
</tr>
<tr class="even">
<td align="left">extractDrugFMF()</td>
<td align="left">FMF descriptor</td>
</tr>
<tr class="odd">
<td align="left">extractDrugFragmentComplexity()</td>
<td align="left">Complexity of a system</td>
</tr>
<tr class="even">
<td align="left">extractDrugGravitationalIndex()</td>
<td align="left">Mass distribution of the molecule</td>
</tr>
<tr class="odd">
<td align="left">extractDrugHBondAcceptorCount()</td>
<td align="left">Number of hydrogen bond acceptors</td>
</tr>
<tr class="even">
<td align="left">extractDrugHBondDonorCount()</td>
<td align="left">Number of hydrogen bond donors</td>
</tr>
<tr class="odd">
<td align="left">extractDrugHybridizationRatio()</td>
<td align="left">Molecular complexity in terms of carbon hybridization states</td>
</tr>
<tr class="even">
<td align="left">extractDrugIPMolecularLearning()</td>
<td align="left">Ionization potential</td>
</tr>
<tr class="odd">
<td align="left">extractDrugKappaShapeIndices()</td>
<td align="left">Kier &amp; Hall Kappa molecular shape indices</td>
</tr>
<tr class="even">
<td align="left">extractDrugKierHallSmarts()</td>
<td align="left">Number of occurrences of the E-State fragments</td>
</tr>
<tr class="odd">
<td align="left">extractDrugLargestChain()</td>
<td align="left">Number of atoms in the largest chain</td>
</tr>
<tr class="even">
<td align="left">extractDrugLargestPiSystem()</td>
<td align="left">Number of atoms in the largest Pi chain</td>
</tr>
<tr class="odd">
<td align="left">extractDrugLengthOverBreadth()</td>
<td align="left">Ratio of length to breadth descriptor</td>
</tr>
<tr class="even">
<td align="left">extractDrugLongestAliphaticChain()</td>
<td align="left">Number of atoms in the longest aliphatic chain</td>
</tr>
<tr class="odd">
<td align="left">extractDrugMannholdLogP()</td>
<td align="left">LogP based on the number of carbons and hetero atoms</td>
</tr>
<tr class="even">
<td align="left">extractDrugMDE()</td>
<td align="left">Molecular Distance Edge (MDE) descriptors for C, N and O</td>
</tr>
<tr class="odd">
<td align="left">extractDrugMomentOfInertia()</td>
<td align="left">Principal moments of inertia and ratios of the principal moments</td>
</tr>
<tr class="even">
<td align="left">extractDrugPetitjeanNumber()</td>
<td align="left">Petitjean number of a molecule</td>
</tr>
<tr class="odd">
<td align="left">extractDrugPetitjeanShapeIndex()</td>
<td align="left">Petitjean shape indices</td>
</tr>
<tr class="even">
<td align="left">extractDrugRotatableBondsCount()</td>
<td align="left">Number of non-rotatable bonds on a molecule</td>
</tr>
<tr class="odd">
<td align="left">extractDrugRuleOfFive()</td>
<td align="left">Number failures of the Lipinskiâ€™s Rule Of Five</td>
</tr>
<tr class="even">
<td align="left">extractDrugTPSA()</td>
<td align="left">Topological Polar Surface Area (TPSA)</td>
</tr>
<tr class="odd">
<td align="left">extractDrugVABC()</td>
<td align="left">Volume of a molecule</td>
</tr>
<tr class="even">
<td align="left">extractDrugVAdjMa()</td>
<td align="left">Vertex adjacency information of a molecule</td>
</tr>
<tr class="odd">
<td align="left">extractDrugWeight()</td>
<td align="left">Total weight of atoms</td>
</tr>
<tr class="even">
<td align="left">extractDrugWeightedPath()</td>
<td align="left">Weighted path (Molecular ID)</td>
</tr>
<tr class="odd">
<td align="left">extractDrugWHIM()</td>
<td align="left">Holistic descriptors described by Todeschini et al.</td>
</tr>
<tr class="even">
<td align="left">extractDrugWienerNumbers()</td>
<td align="left">Wiener path number and wiener polarity number</td>
</tr>
<tr class="odd">
<td align="left">extractDrugXLogP()</td>
<td align="left">Prediction of logP based on the atom-type method called XLogP</td>
</tr>
<tr class="even">
<td align="left">extractDrugZagrebIndex()</td>
<td align="left">Sum of the squared atom degrees of all heavy atoms</td>
</tr>
</tbody>
</table>
</div>
<div id="molecular-fingerprints" class="section level1">
<h1 class="hasAnchor">
<a href="#molecular-fingerprints" class="anchor"></a>Molecular Fingerprints</h1>
<table class="table">
<caption>Table 8: Molecular fingerprints</caption>
<thead><tr class="header">
<th align="left">Function name</th>
<th align="left">Fingerprint type</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">extractDrugStandard()</td>
<td align="left">Standard molecular fingerprints (in compact format)</td>
</tr>
<tr class="even">
<td align="left">extractDrugStandardComplete()</td>
<td align="left">Standard molecular fingerprints (in complete format)</td>
</tr>
<tr class="odd">
<td align="left">extractDrugExtended()</td>
<td align="left">Extended molecular fingerprints (in compact format)</td>
</tr>
<tr class="even">
<td align="left">extractDrugExtendedComplete()</td>
<td align="left">Extended molecular fingerprints (in complete format)</td>
</tr>
<tr class="odd">
<td align="left">extractDrugGraph()</td>
<td align="left">Graph molecular fingerprints (in compact format)</td>
</tr>
<tr class="even">
<td align="left">extractDrugGraphComplete()</td>
<td align="left">Graph molecular fingerprints (in complete format)</td>
</tr>
<tr class="odd">
<td align="left">extractDrugHybridization()</td>
<td align="left">Hybridization molecular fingerprints (in compact format)</td>
</tr>
<tr class="even">
<td align="left">extractDrugHybridizationComplete()</td>
<td align="left">Hybridization molecular fingerprints (in complete format)</td>
</tr>
<tr class="odd">
<td align="left">extractDrugMACCS()</td>
<td align="left">MACCS molecular fingerprints (in compact format)</td>
</tr>
<tr class="even">
<td align="left">extractDrugMACCSComplete()</td>
<td align="left">MACCS molecular fingerprints (in complete format)</td>
</tr>
<tr class="odd">
<td align="left">extractDrugEstate()</td>
<td align="left">E-State molecular fingerprints (in compact format)</td>
</tr>
<tr class="even">
<td align="left">extractDrugEstateComplete()</td>
<td align="left">E-State molecular fingerprints (in complete format)</td>
</tr>
<tr class="odd">
<td align="left">extractDrugPubChem()</td>
<td align="left">PubChem molecular fingerprints (in compact format)</td>
</tr>
<tr class="even">
<td align="left">extractDrugPubChemComplete()</td>
<td align="left">PubChem molecular fingerprints (in complete format)</td>
</tr>
<tr class="odd">
<td align="left">extractDrugKR()</td>
<td align="left">KR (Klekota and Roth) molecular fingerprints (in compact format)</td>
</tr>
<tr class="even">
<td align="left">extractDrugKRComplete()</td>
<td align="left">KR (Klekota and Roth) molecular fingerprints (in complete format)</td>
</tr>
<tr class="odd">
<td align="left">extractDrugShortestPath()</td>
<td align="left">Shortest Path molecular fingerprints (in compact format)</td>
</tr>
<tr class="even">
<td align="left">extractDrugShortestPathComplete()</td>
<td align="left">Shortest Path molecular fingerprints (in complete format)</td>
</tr>
<tr class="odd">
<td align="left">extractDrugOBFP2()</td>
<td align="left">FP2 molecular fingerprints</td>
</tr>
<tr class="even">
<td align="left">extractDrugOBFP3()</td>
<td align="left">FP3 molecular fingerprints</td>
</tr>
<tr class="odd">
<td align="left">extractDrugOBFP4()</td>
<td align="left">FP4 molecular fingerprints</td>
</tr>
<tr class="even">
<td align="left">extractDrugOBMACCS()</td>
<td align="left">MACCS molecular fingerprints</td>
</tr>
</tbody>
</table>
</div>
<div id="protein-protein-and-compound-protein-interation-descriptors" class="section level1">
<h1 class="hasAnchor">
<a href="#protein-protein-and-compound-protein-interation-descriptors" class="anchor"></a>Protein-Protein and Compound-Protein Interation Descriptors</h1>
<table class="table">
<caption>Table 9: Protein-protein and compound-protein interation descriptors</caption>
<thead><tr class="header">
<th align="left">Function name</th>
<th align="left">Function description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">getPPI()</td>
<td align="left">Generating protein-protein interaction descriptors</td>
</tr>
<tr class="even">
<td align="left">getCPI()</td>
<td align="left">Generating compound-protein interaction descriptors</td>
</tr>
</tbody>
</table>
</div>
<div id="similarity-and-similarity-searching" class="section level1">
<h1 class="hasAnchor">
<a href="#similarity-and-similarity-searching" class="anchor"></a>Similarity and Similarity Searching</h1>
<table class="table">
<caption>Table 10: Similarity and similarity searching</caption>
<thead><tr class="header">
<th align="left">Function name</th>
<th align="left">Function description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">calcDrugFPSim()</td>
<td align="left">Calculate drug molecule similarity derived by molecular fingerprints</td>
</tr>
<tr class="even">
<td align="left">calcDrugMCSSim()</td>
<td align="left">Calculate drug molecule similarity derived by maximum common substructure search</td>
</tr>
<tr class="odd">
<td align="left">searchDrug()</td>
<td align="left">Parallelized drug molecule similarity search by molecular fingerprints similarity or maximum common substructure search</td>
</tr>
<tr class="even">
<td align="left">calcTwoProtSeqSim()</td>
<td align="left">Similarity calculation based on sequence alignment for a pair of protein sequences</td>
</tr>
<tr class="odd">
<td align="left">calcParProtSeqSim()</td>
<td align="left">Parallellized protein sequence similarity calculation based on sequence alignment</td>
</tr>
<tr class="even">
<td align="left">calcTwoProtGOSim()</td>
<td align="left">Similarity calculation based on Gene Ontology (GO) similarity between two proteins</td>
</tr>
<tr class="odd">
<td align="left">calcParProtGOSim()</td>
<td align="left">Protein similarity calculation based on Gene Ontology (GO) similarity</td>
</tr>
</tbody>
</table>
</div>
<div id="protein-sequence-data-manipulation" class="section level1">
<h1 class="hasAnchor">
<a href="#protein-sequence-data-manipulation" class="anchor"></a>Protein Sequence Data Manipulation</h1>
<table class="table">
<caption>Table 11: Protein sequence data manipulation</caption>
<thead><tr class="header">
<th align="left">Function name</th>
<th align="left">Function description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">readFASTA()</td>
<td align="left">Read protein sequences in FASTA format</td>
</tr>
<tr class="even">
<td align="left">readPDB()</td>
<td align="left">Read protein sequences in PDB format</td>
</tr>
<tr class="odd">
<td align="left">segProt()</td>
<td align="left">Protein sequence segmentation</td>
</tr>
<tr class="even">
<td align="left">checkProt()</td>
<td align="left">Check if the protein sequenceâ€™s amino acid types are the 20 default types</td>
</tr>
</tbody>
</table>
</div>
<div id="molecular-data-manipulation" class="section level1">
<h1 class="hasAnchor">
<a href="#molecular-data-manipulation" class="anchor"></a>Molecular Data Manipulation</h1>
<table class="table">
<caption>Table 12: Molecular data manipulation</caption>
<thead><tr class="header">
<th align="left">Function name</th>
<th align="left">Function description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">readMolFromSDF()</td>
<td align="left">Read molecules from SDF files and return parsed Java molecular object</td>
</tr>
<tr class="even">
<td align="left">readMolFromSmi()</td>
<td align="left">Read molecules from SMILES files and return parsed Java molecular object or plain text list</td>
</tr>
<tr class="odd">
<td align="left">convMolFormat()</td>
<td align="left">Chemical file formats conversion</td>
</tr>
</tbody>
</table>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#retrieve-protein-sequence-data-from-online-databases">Retrieve Protein Sequence Data from Online Databases</a></li>
      <li><a href="#retrieve-drug-molecular-data-from-online-databases">Retrieve Drug Molecular Data from Online Databases</a></li>
      <li><a href="#calculate-commonly-used-protein-sequence-derived-descriptors">Calculate Commonly Used Protein Sequence Derived Descriptors</a></li>
      <li><a href="#generate-profile-based-protein-representations">Generate Profile-Based Protein Representations</a></li>
      <li><a href="#generate-scales-based-descriptors-for-proteochemometrics-modeling">Generate Scales-Based Descriptors for Proteochemometrics Modeling</a></li>
      <li><a href="#molecular-descriptor-sets-of-the-20-amino-acids-for-generating-scales-based-descriptors">Molecular Descriptor Sets of the 20 Amino Acids for Generating Scales-Based Descriptors</a></li>
      <li><a href="#molecular-descriptors">Molecular Descriptors</a></li>
      <li><a href="#molecular-fingerprints">Molecular Fingerprints</a></li>
      <li><a href="#protein-protein-and-compound-protein-interation-descriptors">Protein-Protein and Compound-Protein Interation Descriptors</a></li>
      <li><a href="#similarity-and-similarity-searching">Similarity and Similarity Searching</a></li>
      <li><a href="#protein-sequence-data-manipulation">Protein Sequence Data Manipulation</a></li>
      <li><a href="#molecular-data-manipulation">Molecular Data Manipulation</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Nan Xiao, Dong-Sheng Cao, Qing-Song Xu.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
