<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Rcpi Quick Reference Card • Rcpi</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Google analytics --><script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-66394392-1', 'auto');
  ga('send', 'pageview');

</script>
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Rcpi</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Rcpi.html">Get Started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Rcpi-quickref.html">Rcpi Quick Reference Card</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/road2stat/Rcpi">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>Rcpi Quick Reference Card</h1>
                        <h4 class="author">Nan Xiao &lt;<a href="https://nanx.me" class="uri">https://nanx.me</a>&gt;</h4>
            
            <h4 class="date">2018-01-09</h4>
          </div>

    
    
<div class="contents">
<section id="retrieve-protein-sequence-data-from-online-databases" class="level1"><h1>Retrieve Protein Sequence Data from Online Databases</h1>
<table class="table">
<caption>Table 1: Retrieving protein sequence data from various online databases</caption>
<thead><tr class="header">
<th style="text-align: left;">Function name</th>
<th style="text-align: left;">Function description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">getProt()</td>
<td style="text-align: left;">Retrieve protein sequence in FASTA format or PDB format from various online databases</td>
</tr>
<tr class="even">
<td style="text-align: left;">getFASTAFromUniProt()</td>
<td style="text-align: left;">Retrieve protein sequence in FASTA format from UniProt</td>
</tr>
<tr class="odd">
<td style="text-align: left;">getFASTAFromKEGG()</td>
<td style="text-align: left;">Retrieve protein sequence in FASTA format from KEGG</td>
</tr>
<tr class="even">
<td style="text-align: left;">getPDBFromRCSBPDB()</td>
<td style="text-align: left;">Retrieve protein sequence in PDB Format from RCSB PDB</td>
</tr>
<tr class="odd">
<td style="text-align: left;">getSeqFromUniProt()</td>
<td style="text-align: left;">Retrieve protein sequence from UniProt</td>
</tr>
<tr class="even">
<td style="text-align: left;">getSeqFromKEGG()</td>
<td style="text-align: left;">Retrieve protein sequence from KEGG</td>
</tr>
<tr class="odd">
<td style="text-align: left;">getSeqFromRCSBPDB()</td>
<td style="text-align: left;">Retrieve protein sequence from RCSB PDB</td>
</tr>
</tbody>
</table></section><section id="retrieve-drug-molecular-data-from-online-databases" class="level1"><h1>Retrieve Drug Molecular Data from Online Databases</h1>
<table class="table">
<caption>Table 2: Retrieving drug molecular data from various online databases</caption>
<thead><tr class="header">
<th style="text-align: left;">Function name</th>
<th style="text-align: left;">Function description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">getDrug()</td>
<td style="text-align: left;">Retrieve drug molecules in MOL format and SMILES format from various online databases</td>
</tr>
<tr class="even">
<td style="text-align: left;">getMolFromDrugBank()</td>
<td style="text-align: left;">Retrieve drug molecules in MOL format from DrugBank</td>
</tr>
<tr class="odd">
<td style="text-align: left;">getMolFromPubChem()</td>
<td style="text-align: left;">Retrieve drug molecules in MOL format from PubChem</td>
</tr>
<tr class="even">
<td style="text-align: left;">getMolFromChEMBL()</td>
<td style="text-align: left;">Retrieve drug molecules in MOL format from ChEMBL</td>
</tr>
<tr class="odd">
<td style="text-align: left;">getMolFromKEGG()</td>
<td style="text-align: left;">Retrieve drug molecules in MOL format from the KEGG</td>
</tr>
<tr class="even">
<td style="text-align: left;">getMolFromCAS()</td>
<td style="text-align: left;">Retrieve drug molecules in InChI format from CAS</td>
</tr>
<tr class="odd">
<td style="text-align: left;">getSmiFromDrugBank()</td>
<td style="text-align: left;">Retrieve drug molecules in SMILES format from DrugBank</td>
</tr>
<tr class="even">
<td style="text-align: left;">getSmiFromPubChem()</td>
<td style="text-align: left;">Retrieve drug molecules in SMILES format from PubChem</td>
</tr>
<tr class="odd">
<td style="text-align: left;">getSmiFromChEMBL()</td>
<td style="text-align: left;">Retrieve drug molecules in SMILES format from ChEMBL</td>
</tr>
<tr class="even">
<td style="text-align: left;">getSmiFromKEGG()</td>
<td style="text-align: left;">Retrieve drug molecules in SMILES format from KEGG</td>
</tr>
</tbody>
</table></section><section id="calculate-commonly-used-protein-sequence-derived-descriptors" class="level1"><h1>Calculate Commonly Used Protein Sequence Derived Descriptors</h1>
<table class="table">
<caption>Table 3: Calculating commonly used protein sequence derived descriptors</caption>
<thead><tr class="header">
<th style="text-align: left;">Function name</th>
<th style="text-align: left;">Descriptor name</th>
<th style="text-align: left;">Descriptor group</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">extractProtAAC()</td>
<td style="text-align: left;">Amino acid composition</td>
<td style="text-align: left;">Amino acid composition</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractProtDC()</td>
<td style="text-align: left;">Dipeptide composition</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractProtTC()</td>
<td style="text-align: left;">Tripeptide composition</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">extractProtMoreauBroto()</td>
<td style="text-align: left;">Normalized Moreau-Broto autocorrelation</td>
<td style="text-align: left;">Autocorrelation</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractProtMoran()</td>
<td style="text-align: left;">Moran autocorrelation</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">extractProtGeary()</td>
<td style="text-align: left;">Geary autocorrelation</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractProtCTDC()</td>
<td style="text-align: left;">Composition</td>
<td style="text-align: left;">CTD</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractProtCTDT()</td>
<td style="text-align: left;">Transition</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractProtCTDD()</td>
<td style="text-align: left;">Distribution</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">extractProtCTriad()</td>
<td style="text-align: left;">Conjoint Triad</td>
<td style="text-align: left;">Conjoint Triad</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractProtSOCN()</td>
<td style="text-align: left;">Sequence-order-coupling number</td>
<td style="text-align: left;">Quasi-sequence-order</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractProtQSO()</td>
<td style="text-align: left;">Quasi-sequence-order descriptors</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractProtPAAC()</td>
<td style="text-align: left;">Pseudo-amino acid composition</td>
<td style="text-align: left;">Pseudo-amino acid composition</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractProtAPAAC()</td>
<td style="text-align: left;">Amphiphilic pseudo-amino acid composition</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">AAindex</td>
<td style="text-align: left;">AAindex data of 544 physicochemical and biological properties for 20 amino acids</td>
<td style="text-align: left;">Dataset</td>
</tr>
</tbody>
</table></section><section id="generate-profile-based-protein-representations" class="level1"><h1>Generate Profile-Based Protein Representations</h1>
<table class="table">
<caption>Table 4: Generating profile-based protein representations</caption>
<thead><tr class="header">
<th style="text-align: left;">Function name</th>
<th style="text-align: left;">Function description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">extractProtPSSM()</td>
<td style="text-align: left;">Compute PSSM (Position-Specific Scoring Matrix) for given protein sequence or peptides</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractProtPSSMFeature()</td>
<td style="text-align: left;">Profile-based protein representation derived by PSSM</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractProtPSSMAcc()</td>
<td style="text-align: left;">Profile-based protein representation derived by PSSM and auto cross covariance (ACC)</td>
</tr>
</tbody>
</table></section><section id="generate-scales-based-descriptors-for-proteochemometrics-modeling" class="level1"><h1>Generate Scales-Based Descriptors for Proteochemometrics Modeling</h1>
<table class="table">
<caption>Table 5: Generating scales-based descriptors for proteochemometrics modeling</caption>
<thead><tr class="header">
<th style="text-align: left;">Function name</th>
<th style="text-align: left;">Descriptor class</th>
<th style="text-align: left;">Derived by</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">extractPCMScales()</td>
<td style="text-align: left;">Generalized scales-based descriptors derived by principal components analysis (PCA)</td>
<td style="text-align: left;">Principal components analysis</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractPCMPropScales()</td>
<td style="text-align: left;">Generalized scales-based descriptors derived by amino acid properties (AAindex)</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractPCMDescScales()</td>
<td style="text-align: left;">Generalized scales-based descriptors derived by 2D and 3D molecular descriptors (Topological, WHIM, VHSE, etc.)</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">extractPCMFAScales()</td>
<td style="text-align: left;">Generalized scales-based descriptors derived by factor analysis</td>
<td style="text-align: left;">Factor analysis</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractPCMMDSScales()</td>
<td style="text-align: left;">Generalized scales-based descriptors derived by multidimensional scaling (MDS)</td>
<td style="text-align: left;">Multidimensional scaling</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractPCMBLOSUM()</td>
<td style="text-align: left;">Generalized BLOSUM and PAM matrix-derived descriptors</td>
<td style="text-align: left;">Substitution matrix</td>
</tr>
<tr class="odd">
<td style="text-align: left;">acc()</td>
<td style="text-align: left;">Auto cross covariance (ACC) for generating scales-based descriptors of the same length</td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table></section><section id="molecular-descriptor-sets-of-the-20-amino-acids-for-generating-scales-based-descriptors" class="level1"><h1>Molecular Descriptor Sets of the 20 Amino Acids for Generating Scales-Based Descriptors</h1>
<table class="table">
<caption>Table 6: Pre-calculated molecular descriptor sets of the 20 amino acids in for generating scales-based descriptors for proteochemometrics modeling.</caption>
<thead><tr class="header">
<th style="text-align: left;">Dataset name</th>
<th style="text-align: left;">Dataset description</th>
<th style="text-align: left;">Dimensionality</th>
<th style="text-align: left;">Calculated by</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">OptAA3d</td>
<td style="text-align: left;">Optimized 20 amino acids</td>
<td style="text-align: left;">–</td>
<td style="text-align: left;">MOE</td>
</tr>
<tr class="even">
<td style="text-align: left;">AA2DACOR</td>
<td style="text-align: left;">2D autocorrelations descriptors</td>
<td style="text-align: left;">92</td>
<td style="text-align: left;">Dragon</td>
</tr>
<tr class="odd">
<td style="text-align: left;">AA3DMoRSE</td>
<td style="text-align: left;">3D-MoRSE descriptors</td>
<td style="text-align: left;">160</td>
<td style="text-align: left;">Dragon</td>
</tr>
<tr class="even">
<td style="text-align: left;">AAACF</td>
<td style="text-align: left;">Atom-centred fragments descriptors</td>
<td style="text-align: left;">6</td>
<td style="text-align: left;">Dragon</td>
</tr>
<tr class="odd">
<td style="text-align: left;">AABurden</td>
<td style="text-align: left;">Burden Eigenvalues descriptors</td>
<td style="text-align: left;">62</td>
<td style="text-align: left;">Dragon</td>
</tr>
<tr class="even">
<td style="text-align: left;">AAConn</td>
<td style="text-align: left;">Connectivity indices descriptors</td>
<td style="text-align: left;">33</td>
<td style="text-align: left;">Dragon</td>
</tr>
<tr class="odd">
<td style="text-align: left;">AAConst</td>
<td style="text-align: left;">Constitutional descriptors</td>
<td style="text-align: left;">23</td>
<td style="text-align: left;">Dragon</td>
</tr>
<tr class="even">
<td style="text-align: left;">AAEdgeAdj</td>
<td style="text-align: left;">Edge adjacency indices descriptors</td>
<td style="text-align: left;">97</td>
<td style="text-align: left;">Dragon</td>
</tr>
<tr class="odd">
<td style="text-align: left;">AAEigIdx</td>
<td style="text-align: left;">Eigenvalue-based indices descriptors</td>
<td style="text-align: left;">44</td>
<td style="text-align: left;">Dragon</td>
</tr>
<tr class="even">
<td style="text-align: left;">AAFGC</td>
<td style="text-align: left;">Functional group counts descriptors</td>
<td style="text-align: left;">5</td>
<td style="text-align: left;">Dragon</td>
</tr>
<tr class="odd">
<td style="text-align: left;">AAGeom</td>
<td style="text-align: left;">Geometrical descriptors</td>
<td style="text-align: left;">41</td>
<td style="text-align: left;">Dragon</td>
</tr>
<tr class="even">
<td style="text-align: left;">AAGETAWAY</td>
<td style="text-align: left;">GETAWAY descriptors</td>
<td style="text-align: left;">194</td>
<td style="text-align: left;">Dragon</td>
</tr>
<tr class="odd">
<td style="text-align: left;">AAInfo</td>
<td style="text-align: left;">Information indices descriptors</td>
<td style="text-align: left;">47</td>
<td style="text-align: left;">Dragon</td>
</tr>
<tr class="even">
<td style="text-align: left;">AAMolProp</td>
<td style="text-align: left;">Molecular properties descriptors</td>
<td style="text-align: left;">12</td>
<td style="text-align: left;">Dragon</td>
</tr>
<tr class="odd">
<td style="text-align: left;">AARandic</td>
<td style="text-align: left;">Randic molecular profiles descriptors</td>
<td style="text-align: left;">41</td>
<td style="text-align: left;">Dragon</td>
</tr>
<tr class="even">
<td style="text-align: left;">AARDF</td>
<td style="text-align: left;">RDF descriptors</td>
<td style="text-align: left;">82</td>
<td style="text-align: left;">Dragon</td>
</tr>
<tr class="odd">
<td style="text-align: left;">AATopo</td>
<td style="text-align: left;">Topological descriptors</td>
<td style="text-align: left;">78</td>
<td style="text-align: left;">Dragon</td>
</tr>
<tr class="even">
<td style="text-align: left;">AATopoChg</td>
<td style="text-align: left;">Topological charge indices descriptors</td>
<td style="text-align: left;">15</td>
<td style="text-align: left;">Dragon</td>
</tr>
<tr class="odd">
<td style="text-align: left;">AAWalk</td>
<td style="text-align: left;">Walk and path counts descriptors</td>
<td style="text-align: left;">40</td>
<td style="text-align: left;">Dragon</td>
</tr>
<tr class="even">
<td style="text-align: left;">AAWHIM</td>
<td style="text-align: left;">WHIM descriptors</td>
<td style="text-align: left;">99</td>
<td style="text-align: left;">Dragon</td>
</tr>
<tr class="odd">
<td style="text-align: left;">AACPSA</td>
<td style="text-align: left;">CPSA descriptors</td>
<td style="text-align: left;">41</td>
<td style="text-align: left;">Accelrys Discovery Studio</td>
</tr>
<tr class="even">
<td style="text-align: left;">AADescAll</td>
<td style="text-align: left;">All the 2D descriptors calculated by Dragon</td>
<td style="text-align: left;">1171</td>
<td style="text-align: left;">Dragon</td>
</tr>
<tr class="odd">
<td style="text-align: left;">AAMOE2D</td>
<td style="text-align: left;">All the 2D descriptors calculated by MOE</td>
<td style="text-align: left;">148</td>
<td style="text-align: left;">MOE</td>
</tr>
<tr class="even">
<td style="text-align: left;">AAMOE3D</td>
<td style="text-align: left;">All the 3D descriptors calculated by MOE</td>
<td style="text-align: left;">143</td>
<td style="text-align: left;">MOE</td>
</tr>
<tr class="odd">
<td style="text-align: left;">AABLOSUM45</td>
<td style="text-align: left;">BLOSUM45 matrix for 20 amino acids</td>
<td style="text-align: left;"><span class="math inline">\(20 \times 20\)</span></td>
<td style="text-align: left;">Biostrings</td>
</tr>
<tr class="even">
<td style="text-align: left;">AABLOSUM50</td>
<td style="text-align: left;">BLOSUM50 matrix for 20 amino acids</td>
<td style="text-align: left;"><span class="math inline">\(20 \times 20\)</span></td>
<td style="text-align: left;">Biostrings</td>
</tr>
<tr class="odd">
<td style="text-align: left;">AABLOSUM62</td>
<td style="text-align: left;">BLOSUM62 matrix for 20 amino acids</td>
<td style="text-align: left;"><span class="math inline">\(20 \times 20\)</span></td>
<td style="text-align: left;">Biostrings</td>
</tr>
<tr class="even">
<td style="text-align: left;">AABLOSUM80</td>
<td style="text-align: left;">BLOSUM80 matrix for 20 amino acids</td>
<td style="text-align: left;"><span class="math inline">\(20 \times 20\)</span></td>
<td style="text-align: left;">Biostrings</td>
</tr>
<tr class="odd">
<td style="text-align: left;">AABLOSUM100</td>
<td style="text-align: left;">BLOSUM100 matrix for 20 amino acids</td>
<td style="text-align: left;"><span class="math inline">\(20 \times 20\)</span></td>
<td style="text-align: left;">Biostrings</td>
</tr>
<tr class="even">
<td style="text-align: left;">AAPAM30</td>
<td style="text-align: left;">PAM30 matrix for 20 amino acids</td>
<td style="text-align: left;"><span class="math inline">\(20 \times 20\)</span></td>
<td style="text-align: left;">Biostrings</td>
</tr>
<tr class="odd">
<td style="text-align: left;">AAPAM40</td>
<td style="text-align: left;">PAM40 matrix for 20 amino acids</td>
<td style="text-align: left;"><span class="math inline">\(20 \times 20\)</span></td>
<td style="text-align: left;">Biostrings</td>
</tr>
<tr class="even">
<td style="text-align: left;">AAPAM70</td>
<td style="text-align: left;">PAM70 matrix for 20 amino acids</td>
<td style="text-align: left;"><span class="math inline">\(20 \times 20\)</span></td>
<td style="text-align: left;">Biostrings</td>
</tr>
<tr class="odd">
<td style="text-align: left;">AAPAM120</td>
<td style="text-align: left;">PAM120 matrix for 20 amino acids</td>
<td style="text-align: left;"><span class="math inline">\(20 \times 20\)</span></td>
<td style="text-align: left;">Biostrings</td>
</tr>
<tr class="even">
<td style="text-align: left;">AAPAM250</td>
<td style="text-align: left;">PAM250 matrix for 20 amino acids</td>
<td style="text-align: left;"><span class="math inline">\(20 \times 20\)</span></td>
<td style="text-align: left;">Biostrings</td>
</tr>
</tbody>
</table>
<p><strong>Note</strong>: non-informative descriptors (e.g. descriptors with only one value across all the 20 amino acids) in these datasets have been filtered out.</p>
</section><section id="molecular-descriptors" class="level1"><h1>Molecular Descriptors</h1>
<table class="table">
<caption>Table 7: Molecular descriptors</caption>
<thead><tr class="header">
<th style="text-align: left;">Function name</th>
<th style="text-align: left;">Descriptor name</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">extractDrugAIO()</td>
<td style="text-align: left;">All the molecular descriptors in the package</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugALOGP()</td>
<td style="text-align: left;">Atom additive logP and molar refractivity values descriptor</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugAminoAcidCount()</td>
<td style="text-align: left;">Number of amino acids</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugApol()</td>
<td style="text-align: left;">Sum of the atomic polarizabilities</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugAromaticAtomsCount()</td>
<td style="text-align: left;">Number of aromatic atoms</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugAromaticBondsCount()</td>
<td style="text-align: left;">Number of aromatic bonds</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugAtomCount()</td>
<td style="text-align: left;">Number of atom descriptor</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugAutocorrelationCharge()</td>
<td style="text-align: left;">Moreau-Broto autocorrelation descriptors using partial charges</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugAutocorrelationMass()</td>
<td style="text-align: left;">Moreau-Broto autocorrelation descriptors using atomic weight</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugAutocorrelationPolarizability()</td>
<td style="text-align: left;">Moreau-Broto autocorrelation descriptors using polarizability</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugBCUT()</td>
<td style="text-align: left;">BCUT, the eigenvalue based descriptor</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugBondCount()</td>
<td style="text-align: left;">Number of bonds of a certain bond order</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugBPol()</td>
<td style="text-align: left;">Sum of the absolute value of the difference between atomic polarizabilities of all bonded atoms in the molecule</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugCarbonTypes()</td>
<td style="text-align: left;">Topological descriptor characterizing the carbon connectivity in terms of hybridization</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugChiChain()</td>
<td style="text-align: left;">Kier &amp; Hall Chi chain indices of orders 3, 4, 5, 6 and 7</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugChiCluster()</td>
<td style="text-align: left;">Kier &amp; Hall Chi cluster indices of orders 3, 4, 5 and 6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugChiPath()</td>
<td style="text-align: left;">Kier &amp; Hall Chi path indices of orders 0 to 7</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugChiPathCluster()</td>
<td style="text-align: left;">Kier &amp; Hall Chi path cluster indices of orders 4, 5 and 6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugCPSA()</td>
<td style="text-align: left;">Descriptors combining surface area and partial charge information</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugDescOB()</td>
<td style="text-align: left;">Molecular descriptors provided by OpenBabel</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugECI()</td>
<td style="text-align: left;">Eccentric connectivity index descriptor</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugFMF()</td>
<td style="text-align: left;">FMF descriptor</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugFragmentComplexity()</td>
<td style="text-align: left;">Complexity of a system</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugGravitationalIndex()</td>
<td style="text-align: left;">Mass distribution of the molecule</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugHBondAcceptorCount()</td>
<td style="text-align: left;">Number of hydrogen bond acceptors</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugHBondDonorCount()</td>
<td style="text-align: left;">Number of hydrogen bond donors</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugHybridizationRatio()</td>
<td style="text-align: left;">Molecular complexity in terms of carbon hybridization states</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugIPMolecularLearning()</td>
<td style="text-align: left;">Ionization potential</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugKappaShapeIndices()</td>
<td style="text-align: left;">Kier &amp; Hall Kappa molecular shape indices</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugKierHallSmarts()</td>
<td style="text-align: left;">Number of occurrences of the E-State fragments</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugLargestChain()</td>
<td style="text-align: left;">Number of atoms in the largest chain</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugLargestPiSystem()</td>
<td style="text-align: left;">Number of atoms in the largest Pi chain</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugLengthOverBreadth()</td>
<td style="text-align: left;">Ratio of length to breadth descriptor</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugLongestAliphaticChain()</td>
<td style="text-align: left;">Number of atoms in the longest aliphatic chain</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugMannholdLogP()</td>
<td style="text-align: left;">LogP based on the number of carbons and hetero atoms</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugMDE()</td>
<td style="text-align: left;">Molecular Distance Edge (MDE) descriptors for C, N and O</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugMomentOfInertia()</td>
<td style="text-align: left;">Principal moments of inertia and ratios of the principal moments</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugPetitjeanNumber()</td>
<td style="text-align: left;">Petitjean number of a molecule</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugPetitjeanShapeIndex()</td>
<td style="text-align: left;">Petitjean shape indices</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugRotatableBondsCount()</td>
<td style="text-align: left;">Number of non-rotatable bonds on a molecule</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugRuleOfFive()</td>
<td style="text-align: left;">Number failures of the Lipinski’s Rule Of Five</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugTPSA()</td>
<td style="text-align: left;">Topological Polar Surface Area (TPSA)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugVABC()</td>
<td style="text-align: left;">Volume of a molecule</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugVAdjMa()</td>
<td style="text-align: left;">Vertex adjacency information of a molecule</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugWeight()</td>
<td style="text-align: left;">Total weight of atoms</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugWeightedPath()</td>
<td style="text-align: left;">Weighted path (Molecular ID)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugWHIM()</td>
<td style="text-align: left;">Holistic descriptors described by Todeschini et al.</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugWienerNumbers()</td>
<td style="text-align: left;">Wiener path number and wiener polarity number</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugXLogP()</td>
<td style="text-align: left;">Prediction of logP based on the atom-type method called XLogP</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugZagrebIndex()</td>
<td style="text-align: left;">Sum of the squared atom degrees of all heavy atoms</td>
</tr>
</tbody>
</table></section><section id="molecular-fingerprints" class="level1"><h1>Molecular Fingerprints</h1>
<table class="table">
<caption>Table 8: Molecular fingerprints</caption>
<thead><tr class="header">
<th style="text-align: left;">Function name</th>
<th style="text-align: left;">Fingerprint type</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">extractDrugStandard()</td>
<td style="text-align: left;">Standard molecular fingerprints (in compact format)</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugStandardComplete()</td>
<td style="text-align: left;">Standard molecular fingerprints (in complete format)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugExtended()</td>
<td style="text-align: left;">Extended molecular fingerprints (in compact format)</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugExtendedComplete()</td>
<td style="text-align: left;">Extended molecular fingerprints (in complete format)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugGraph()</td>
<td style="text-align: left;">Graph molecular fingerprints (in compact format)</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugGraphComplete()</td>
<td style="text-align: left;">Graph molecular fingerprints (in complete format)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugHybridization()</td>
<td style="text-align: left;">Hybridization molecular fingerprints (in compact format)</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugHybridizationComplete()</td>
<td style="text-align: left;">Hybridization molecular fingerprints (in complete format)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugMACCS()</td>
<td style="text-align: left;">MACCS molecular fingerprints (in compact format)</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugMACCSComplete()</td>
<td style="text-align: left;">MACCS molecular fingerprints (in complete format)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugEstate()</td>
<td style="text-align: left;">E-State molecular fingerprints (in compact format)</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugEstateComplete()</td>
<td style="text-align: left;">E-State molecular fingerprints (in complete format)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugPubChem()</td>
<td style="text-align: left;">PubChem molecular fingerprints (in compact format)</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugPubChemComplete()</td>
<td style="text-align: left;">PubChem molecular fingerprints (in complete format)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugKR()</td>
<td style="text-align: left;">KR (Klekota and Roth) molecular fingerprints (in compact format)</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugKRComplete()</td>
<td style="text-align: left;">KR (Klekota and Roth) molecular fingerprints (in complete format)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugShortestPath()</td>
<td style="text-align: left;">Shortest Path molecular fingerprints (in compact format)</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugShortestPathComplete()</td>
<td style="text-align: left;">Shortest Path molecular fingerprints (in complete format)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugOBFP2()</td>
<td style="text-align: left;">FP2 molecular fingerprints</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugOBFP3()</td>
<td style="text-align: left;">FP3 molecular fingerprints</td>
</tr>
<tr class="odd">
<td style="text-align: left;">extractDrugOBFP4()</td>
<td style="text-align: left;">FP4 molecular fingerprints</td>
</tr>
<tr class="even">
<td style="text-align: left;">extractDrugOBMACCS()</td>
<td style="text-align: left;">MACCS molecular fingerprints</td>
</tr>
</tbody>
</table></section><section id="protein-protein-and-compound-protein-interation-descriptors" class="level1"><h1>Protein-Protein and Compound-Protein Interation Descriptors</h1>
<table class="table">
<caption>Table 9: Protein-protein and compound-protein interation descriptors</caption>
<thead><tr class="header">
<th style="text-align: left;">Function name</th>
<th style="text-align: left;">Function description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">getPPI()</td>
<td style="text-align: left;">Generating protein-protein interaction descriptors</td>
</tr>
<tr class="even">
<td style="text-align: left;">getCPI()</td>
<td style="text-align: left;">Generating compound-protein interaction descriptors</td>
</tr>
</tbody>
</table></section><section id="similarity-and-similarity-searching" class="level1"><h1>Similarity and Similarity Searching</h1>
<table class="table">
<caption>Table 10: Similarity and similarity searching</caption>
<thead><tr class="header">
<th style="text-align: left;">Function name</th>
<th style="text-align: left;">Function description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">calcDrugFPSim()</td>
<td style="text-align: left;">Calculate drug molecule similarity derived by molecular fingerprints</td>
</tr>
<tr class="even">
<td style="text-align: left;">calcDrugMCSSim()</td>
<td style="text-align: left;">Calculate drug molecule similarity derived by maximum common substructure search</td>
</tr>
<tr class="odd">
<td style="text-align: left;">searchDrug()</td>
<td style="text-align: left;">Parallelized drug molecule similarity search by molecular fingerprints similarity or maximum common substructure search</td>
</tr>
<tr class="even">
<td style="text-align: left;">calcTwoProtSeqSim()</td>
<td style="text-align: left;">Similarity calculation based on sequence alignment for a pair of protein sequences</td>
</tr>
<tr class="odd">
<td style="text-align: left;">calcParProtSeqSim()</td>
<td style="text-align: left;">Parallellized protein sequence similarity calculation based on sequence alignment</td>
</tr>
<tr class="even">
<td style="text-align: left;">calcTwoProtGOSim()</td>
<td style="text-align: left;">Similarity calculation based on Gene Ontology (GO) similarity between two proteins</td>
</tr>
<tr class="odd">
<td style="text-align: left;">calcParProtGOSim()</td>
<td style="text-align: left;">Protein similarity calculation based on Gene Ontology (GO) similarity</td>
</tr>
</tbody>
</table></section><section id="protein-sequence-data-manipulation" class="level1"><h1>Protein Sequence Data Manipulation</h1>
<table class="table">
<caption>Table 11: Protein sequence data manipulation</caption>
<thead><tr class="header">
<th style="text-align: left;">Function name</th>
<th style="text-align: left;">Function description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">readFASTA()</td>
<td style="text-align: left;">Read protein sequences in FASTA format</td>
</tr>
<tr class="even">
<td style="text-align: left;">readPDB()</td>
<td style="text-align: left;">Read protein sequences in PDB format</td>
</tr>
<tr class="odd">
<td style="text-align: left;">segProt()</td>
<td style="text-align: left;">Protein sequence segmentation</td>
</tr>
<tr class="even">
<td style="text-align: left;">checkProt()</td>
<td style="text-align: left;">Check if the protein sequence’s amino acid types are the 20 default types</td>
</tr>
</tbody>
</table></section><section id="molecular-data-manipulation" class="level1"><h1>Molecular Data Manipulation</h1>
<table class="table">
<caption>Table 12: Molecular data manipulation</caption>
<thead><tr class="header">
<th style="text-align: left;">Function name</th>
<th style="text-align: left;">Function description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">readMolFromSDF()</td>
<td style="text-align: left;">Read molecules from SDF files and return parsed Java molecular object</td>
</tr>
<tr class="even">
<td style="text-align: left;">readMolFromSmi()</td>
<td style="text-align: left;">Read molecules from SMILES files and return parsed Java molecular object or plain text list</td>
</tr>
<tr class="odd">
<td style="text-align: left;">convMolFormat()</td>
<td style="text-align: left;">Chemical file formats conversion</td>
</tr>
</tbody>
</table></section>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#retrieve-protein-sequence-data-from-online-databases">Retrieve Protein Sequence Data from Online Databases</a></li>
      <li><a href="#retrieve-drug-molecular-data-from-online-databases">Retrieve Drug Molecular Data from Online Databases</a></li>
      <li><a href="#calculate-commonly-used-protein-sequence-derived-descriptors">Calculate Commonly Used Protein Sequence Derived Descriptors</a></li>
      <li><a href="#generate-profile-based-protein-representations">Generate Profile-Based Protein Representations</a></li>
      <li><a href="#generate-scales-based-descriptors-for-proteochemometrics-modeling">Generate Scales-Based Descriptors for Proteochemometrics Modeling</a></li>
      <li><a href="#molecular-descriptor-sets-of-the-20-amino-acids-for-generating-scales-based-descriptors">Molecular Descriptor Sets of the 20 Amino Acids for Generating Scales-Based Descriptors</a></li>
      <li><a href="#molecular-descriptors">Molecular Descriptors</a></li>
      <li><a href="#molecular-fingerprints">Molecular Fingerprints</a></li>
      <li><a href="#protein-protein-and-compound-protein-interation-descriptors">Protein-Protein and Compound-Protein Interation Descriptors</a></li>
      <li><a href="#similarity-and-similarity-searching">Similarity and Similarity Searching</a></li>
      <li><a href="#protein-sequence-data-manipulation">Protein Sequence Data Manipulation</a></li>
      <li><a href="#molecular-data-manipulation">Molecular Data Manipulation</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Nan Xiao, Dong-Sheng Cao, Qing-Song Xu.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
